roachtest: cdc/tpcc-1000/rangefeed=true failed

SHA: https://github.com/cockroachdb/cockroach/commits/c0cab6082c2383e77f0a806c10a1506b6aa7b6bf

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=cdc/tpcc-1000/rangefeed=true PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1525981&tab=artifacts#/cdc/tpcc-1000/rangefeed=true

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/20191008-1525981/cdc/tpcc-1000/rangefeed=true/run_1
	cluster.go:1764,cdc.go:749,cdc.go:136,cluster.go:2120,errgroup.go:57: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1570511373-26-n4cpu16:4 -- ./workload run tpcc --warehouses=1000 --duration=120m --tolerate-errors {pgurl:1-3}  returned:
		stderr:
		
		stdout:
		    8.4  36507.2  45097.2  49392.1  49392.1 delivery
		   67.0s     2257          238.8           46.7  49392.1  49392.1  49392.1  49392.1 newOrder
		   67.0s     2257           11.0            9.1  38654.7  40802.2  40802.2  40802.2 orderStatus
		   67.0s     2257           96.9           85.4  40802.2  47244.6  55834.6  60129.5 payment
		   67.0s     2257           11.0            9.4  36507.2  40802.2  47244.6  47244.6 stockLevel
		E191008 06:22:35.616065 1 workload/cli/run.go:444  error in newOrder: dial tcp 10.128.0.70:26257: connect: connection refused
		   68.0s     2381           18.0            8.6  38654.7  45097.2  66572.0  66572.0 delivery
		   68.0s     2381          272.8           50.1  42949.7  51539.6  51539.6  51539.6 newOrder
		   68.0s     2381           12.0            9.1  33286.0  40802.2  40802.2  40802.2 orderStatus
		   68.0s     2381          147.9           86.3  38654.7  64424.5  66572.0  66572.0 payment
		   68.0s     2381           10.0            9.4  34359.7  38654.7  38654.7  38654.7 stockLevel
		: signal: killed

```

SHA: https://github.com/cockroachdb/cockroach/commits/239513342a2d23f683bbc1d386f87ff59cc78d10

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=cdc/tpcc-1000/rangefeed=true PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1575479&tab=artifacts#/cdc/tpcc-1000/rangefeed=true

```
The test failed on branch=provisional_201910141814_v19.2.0-rc.1, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/20191106-1575479/cdc/tpcc-1000/rangefeed=true/run_1
	cluster.go:1783,cdc.go:749,cdc.go:136,cluster.go:2142,errgroup.go:57: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1575479-1573004318-25-n4cpu16:4 -- ./workload run tpcc --warehouses=1000 --duration=120m --tolerate-errors {pgurl:1-3}  returned:
		stderr:
		
		stdout:
		    8.7  30064.8  34359.7  34359.7  34359.7 delivery
		   43.0s     1373            0.0            0.0      0.0      0.0      0.0      0.0 newOrder
		   43.0s     1373            3.0           10.3  26843.5  26843.5  26843.5  26843.5 orderStatus
		   43.0s     1373          657.3           63.6  40802.2  40802.2  40802.2  40802.2 payment
		   43.0s     1373            3.0           10.0  32212.3  34359.7  34359.7  34359.7 stockLevel
		E191106 03:07:47.965221 1 workload/cli/run.go:444  error in delivery: dial tcp 10.128.0.31:26257: connect: connection refused
		   44.0s     1563            4.0            8.5  42949.7  42949.7  42949.7  42949.7 delivery
		   44.0s     1563            0.0            0.0      0.0      0.0      0.0      0.0 newOrder
		   44.0s     1563            4.0           10.2  36507.2  36507.2  36507.2  36507.2 orderStatus
		   44.0s     1563          181.8           66.3  40802.2  42949.7  42949.7  42949.7 payment
		   44.0s     1563            1.0            9.8  32212.3  32212.3  32212.3  32212.3 stockLevel
		: context canceled

```

SHA: https://github.com/cockroachdb/cockroach/commits/35e138aa3c2be545fb4e17a85ea6f1b8d6525e53

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=cdc/tpcc-1000/rangefeed=true PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1587967&tab=artifacts#/cdc/tpcc-1000/rangefeed=true

```
The test failed on branch=provisional_201911111508_v20.1.0-alpha.20191118, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/20191112-1587967/cdc/tpcc-1000/rangefeed=true/run_1
	cluster.go:1783,cdc.go:749,cdc.go:136,cluster.go:2142,errgroup.go:57: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1587967-1573572266-25-n4cpu16:4 -- ./workload run tpcc --warehouses=1000 --duration=120m --tolerate-errors {pgurl:1-3}  returned:
		stderr:
		
		stdout:
		.0           80.2      0.0      0.0      0.0      0.0 newOrder
		   52.0s     2387            0.0           13.2      0.0      0.0      0.0      0.0 orderStatus
		   52.0s     2387            0.0          118.5      0.0      0.0      0.0      0.0 payment
		   52.0s     2387            0.0           12.4      0.0      0.0      0.0      0.0 stockLevel
		E191112 16:39:42.357855 1 workload/cli/run.go:444  error in payment: dial tcp 10.128.0.59:26257: connect: connection refused
		_elapsed___errors__ops/sec(inst)___ops/sec(cum)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)
		   53.0s     2533            0.0           12.0      0.0      0.0      0.0      0.0 delivery
		   53.0s     2533            0.0           78.7      0.0      0.0      0.0      0.0 newOrder
		   53.0s     2533            0.0           12.9      0.0      0.0      0.0      0.0 orderStatus
		   53.0s     2533            0.0          116.3      0.0      0.0      0.0      0.0 payment
		   53.0s     2533            0.0           12.1      0.0      0.0      0.0      0.0 stockLevel
		: context canceled

```

[(roachtest).cdc/tpcc-1000/rangefeed=true failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1642926&tab=buildLog) on [provisional_201912132008_v20.1.0-alpha20191216@e9e2a80361a25fd9f9b179f84be4c5c3d7e7d8cb](https://github.com/cockroachdb/cockroach/commits/e9e2a80361a25fd9f9b179f84be4c5c3d7e7d8cb):

```
The test failed on branch=provisional_201912132008_v20.1.0-alpha20191216, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/20191213-1642926/cdc/tpcc-1000/rangefeed=true/run_1
	cluster.go:1783,cdc.go:785,cdc.go:136,cluster.go:2142,errgroup.go:57: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1642926-1576273385-25-n4cpu16:4 -- ./workload run tpcc --warehouses=1000 --duration=120m --tolerate-errors {pgurl:1-3}  returned:
		stderr:
		
		stdout:
		5  64424.5 newOrder
		   81.0s     2252           24.0            8.4  38654.7  47244.6  47244.6  47244.6 orderStatus
		   81.0s     2252          246.0           70.8  47244.6  62277.0  77309.4  81604.4 payment
		   81.0s     2252           24.0            7.7  42949.7  51539.6  51539.6  51539.6 stockLevel
		E191213 23:03:52.755742 1 workload/cli/run.go:444  error in payment: dial tcp 10.128.0.129:26257: connect: connection refused
		   82.0s     2428           20.0            7.2  40802.2  53687.1  55834.6  55834.6 delivery
		   82.0s     2428          449.2           46.4  45097.2  64424.5  64424.5  64424.5 newOrder
		   82.0s     2428           18.0            8.5  34359.7  40802.2  40802.2  40802.2 orderStatus
		   82.0s     2428          186.7           72.2  42949.7  60129.5  81604.4  81604.4 payment
		   82.0s     2428           17.0            7.8  36507.2  47244.6  49392.1  49392.1 stockLevel
		E191213 23:03:53.756334 1 workload/cli/run.go:444  error in payment: dial tcp 10.128.0.129:26257: connect: connection refused
		: context canceled
```

<details><summary>Repro</summary><p>


Artifacts: [/cdc/tpcc-1000/rangefeed=true](https://teamcity.cockroachdb.com/viewLog.html?buildId=1642926&tab=artifacts#/cdc/tpcc-1000/rangefeed=true)
```
make stressrace TESTS=cdc/tpcc-1000/rangefeed=true PKG=./pkg/roachtest TESTTIMEOUT=5m STRESSFLAGS=-timeout 5m' 2>&1
```

<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).cdc/tpcc-1000/rangefeed=true failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1643204&tab=buildLog) on [release-19.2@2affe9ca6449d959e10efabacba6aa05d7ff7a3a](https://github.com/cockroachdb/cockroach/commits/2affe9ca6449d959e10efabacba6aa05d7ff7a3a):

```
The test failed on branch=release-19.2, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/20191214-1643204/cdc/tpcc-1000/rangefeed=true/run_1
	cluster.go:1783,cdc.go:785,cdc.go:136,cluster.go:2142,errgroup.go:57: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1643204-1576308775-24-n4cpu16:4 -- ./workload run tpcc --warehouses=1000 --duration=120m --tolerate-errors {pgurl:1-3}  returned:
		stderr:
		
		stdout:
		    7.4  51539.6  57982.1  57982.1  57982.1 delivery
		   87.0s     2543          150.0           47.5  53687.1  73014.4  73014.4  73014.4 newOrder
		   87.0s     2543           25.0            7.1  49392.1  60129.5  62277.0  62277.0 orderStatus
		   87.0s     2543          151.0           67.2  53687.1  68719.5  85899.3  85899.3 payment
		   87.0s     2543           17.0            7.4  51539.6  60129.5  60129.5  60129.5 stockLevel
		E191214 08:43:51.482203 1 workload/cli/run.go:444  error in payment: dial tcp 10.128.0.104:26257: connect: connection refused
		   88.0s     2791           10.0            7.4  51539.6  62277.0  62277.0  62277.0 delivery
		   88.0s     2791          178.0           48.9  51539.6  73014.4  73014.4  73014.4 newOrder
		   88.0s     2791           17.0            7.2  47244.6  51539.6  53687.1  53687.1 orderStatus
		   88.0s     2791          140.0           68.0  51539.6  85899.3  85899.3  85899.3 payment
		   88.0s     2791           20.0            7.6  45097.2  57982.1  64424.5  64424.5 stockLevel
		: context canceled
```

<details><summary>Repro</summary><p>


Artifacts: [/cdc/tpcc-1000/rangefeed=true](https://teamcity.cockroachdb.com/viewLog.html?buildId=1643204&tab=artifacts#/cdc/tpcc-1000/rangefeed=true)
```
make stressrace TESTS=cdc/tpcc-1000/rangefeed=true PKG=./pkg/roachtest TESTTIMEOUT=5m STRESSFLAGS=-timeout 5m' 2>&1
```

<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).cdc/tpcc-1000/rangefeed=true failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1683261&tab=buildLog) on [release-19.2@486a8ac25e0470211f6f7659b26c82950ed013be](https://github.com/cockroachdb/cockroach/commits/486a8ac25e0470211f6f7659b26c82950ed013be):

```
The test failed on branch=release-19.2, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/20200110-1683261/cdc/tpcc-1000/rangefeed=true/run_1
	cluster.go:1809,cdc.go:791,cdc.go:136,cluster.go:2168,errgroup.go:57: error with attached stack trace:
		    main.execCmd
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:399
		    main.(*cluster).RunL
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:1843
		    main.(*cluster).Run
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:1807
		    main.(*tpccWorkload).run
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:791
		    main.cdcBasicTest.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:136
		    main.(*monitor).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2168
		    github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - error with embedded safe details: %s returned:
		    stderr:
		    %s
		    stdout:
		    %s
		    -- arg 1: <string>
		    -- arg 2: <string>
		    -- arg 3: <string>
		  - /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1683261-1578641619-24-n4cpu16:4 -- ./workload run tpcc --warehouses=1000 --duration=120m --tolerate-errors {pgurl:1-3}  returned:
		    stderr:
		    
		    stdout:
		           0           17.0           19.3    503.3   1208.0   1476.4   1476.4 stockLevel
		     7179.0s        0           23.0           19.3   2147.5   6710.9   9126.8   9126.8 delivery
		     7179.0s        0          189.1          192.6   2147.5   4563.4   6710.9   9663.7 newOrder
		     7179.0s        0           21.0           19.3    503.3   1140.9   1342.2   1342.2 orderStatus
		     7179.0s        0          192.1          192.8   1342.2   2550.1   3489.7   4563.4 payment
		     7179.0s        0           14.0           19.3    335.5    973.1   1140.9   1140.9 stockLevel
		     7180.0s        0           16.0           19.3   2415.9   6174.0   6442.5   6442.5 delivery
		     7180.0s        0          205.3          192.6   2415.9   4831.8   6442.5   7516.2 newOrder
		     7180.0s        0           18.0           19.3    469.8   1208.0   1208.0   1208.0 orderStatus
		     7180.0s        0          176.3          192.8   1342.2   2818.6   4026.5   9663.7 payment
		     7180.0s        0           17.0           19.3    906.0   1275.1   2281.7   2281.7 stockLevel:
		  - context canceled
```

<details><summary>Repro</summary><p>


Artifacts: [/cdc/tpcc-1000/rangefeed=true](https://teamcity.cockroachdb.com/viewLog.html?buildId=1683261&tab=artifacts#/cdc/tpcc-1000/rangefeed=true)
```
make stressrace TESTS=cdc/tpcc-1000/rangefeed=true PKG=./pkg/roachtest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```

<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).cdc/tpcc-1000/rangefeed=true failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1688037&tab=buildLog) on [release-19.2@7f2413b36fe1ea31a08c3ddd0d1e4b1cd006db37](https://github.com/cockroachdb/cockroach/commits/7f2413b36fe1ea31a08c3ddd0d1e4b1cd006db37):

```
		    -- arg 1: <string>
		    -- arg 2: <string>
		    -- arg 3: <string>
		  - /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1688037-1578987224-25-n4cpu16:4 -- ./workload run tpcc --warehouses=1000 --duration=120m --tolerate-errors {pgurl:1-3}  returned:
		    stderr:
		    
		    stdout:
		               0.0            0.0      0.0      0.0      0.0      0.0 newOrder
		       43.0s     1341            0.0           11.1      0.0      0.0      0.0      0.0 orderStatus
		       43.0s     1341            3.0           46.4  40802.2  40802.2  40802.2  40802.2 payment
		       43.0s     1341            0.0           10.1      0.0      0.0      0.0      0.0 stockLevel
		    E200114 08:23:56.731572 1 workload/cli/run.go:444  error in newOrder: dial tcp 10.128.0.69:26257: connect: connection refused
		       44.0s     1452            5.0            9.3  33286.0  42949.7  42949.7  42949.7 delivery
		       44.0s     1452            0.0            0.0      0.0      0.0      0.0      0.0 newOrder
		       44.0s     1452            6.0           11.0  32212.3  33286.0  33286.0  33286.0 orderStatus
		       44.0s     1452          406.6           54.6  40802.2  42949.7  42949.7  42949.7 payment
		       44.0s     1452            3.0            9.9  31138.5  33286.0  33286.0  33286.0 stockLevel
		    E200114 08:23:57.830231 1 workload/cli/run.go:444  error in newOrder: dial tcp 10.128.0.69:26257: connect: connection refused
		    _elapsed___errors__ops/sec(inst)___ops/sec(cum)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)
		       45.0s     1636            6.0            9.3  30064.8  36507.2  36507.2  36507.2 delivery
		       45.0s     1636            0.0            0.0      0.0      0.0      0.0      0.0 newOrder
		       45.0s     1636           10.0           11.0  31138.5  34359.7  34359.7  34359.7 orderStatus
		       45.0s     1636          533.2           65.3  42949.7  42949.7  42949.7  42949.7 payment
		       45.0s     1636            7.0            9.8  31138.5  36507.2  36507.2  36507.2 stockLevel
		       46.0s     2239            0.0            9.1      0.0      0.0      0.0      0.0 delivery
		       46.0s     2239            0.0            0.0      0.0      0.0      0.0      0.0 newOrder
		       46.0s     2239            0.0           10.7      0.0      0.0      0.0      0.0 orderStatus
		       46.0s     2239           47.2           64.9  42949.7  42949.7  45097.2  45097.2 payment
		       46.0s     2239            0.0            9.6      0.0      0.0      0.0      0.0 stockLevel
		    E200114 08:23:59.012751 1 workload/cli/run.go:444  error in stockLevel: dial tcp 10.128.0.69:26257: connect: connection refused
		       47.0s     2282            3.0            8.9  42949.7  45097.2  45097.2  45097.2 delivery
		       47.0s     2282            0.0            0.0      0.0      0.0      0.0      0.0 newOrder
		       47.0s     2282            6.0           10.6  33286.0  34359.7  34359.7  34359.7 orderStatus
		       47.0s     2282          306.5           70.0  45097.2  45097.2  45097.2  45097.2 payment
		       47.0s     2282            2.0            9.5  34359.7  38654.7  38654.7  38654.7 stockLevel
		    E200114 08:24:00.048460 1 workload/cli/run.go:444  error in payment: dial tcp 10.128.0.69:26257: connect: connection refused
		       48.0s     2443            6.0            8.9  36507.2  45097.2  45097.2  45097.2 delivery
		       48.0s     2443            0.0            0.0      0.0      0.0      0.0      0.0 newOrder
		       48.0s     2443            6.0           10.5  31138.5  33286.0  33286.0  33286.0 orderStatus
		       48.0s     2443          237.0           73.5  45097.2  45097.2  45097.2  45097.2 payment
		       48.0s     2443            5.0            9.4  36507.2  36507.2  36507.2  36507.2 stockLevel
		    E200114 08:24:01.075629 1 workload/cli/run.go:444  error in payment: dial tcp 10.128.0.69:26257: connect: connection refused
		    _elapsed___errors__ops/sec(inst)___ops/sec(cum)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)
		       49.0s     2703            8.0            8.9  38654.7  47244.6  47244.6  47244.6 delivery
		       49.0s     2703          171.2            3.5  31138.5  31138.5  31138.5  31138.5 newOrder
		       49.0s     2703            2.0           10.3  32212.3  33286.0  33286.0  33286.0 orderStatus
		       49.0s     2703          152.2           75.1  47244.6  47244.6  47244.6  47244.6 payment
		       49.0s     2703            7.0            9.3  40802.2  42949.7  42949.7  42949.7 stockLevel
		    E200114 08:24:02.088723 1 workload/cli/run.go:444  error in newOrder: dial tcp 10.128.0.69:26257: connect: connection refused:
		  - context canceled
```

<details><summary>Repro</summary><p>


Artifacts: [/cdc/tpcc-1000/rangefeed=true](https://teamcity.cockroachdb.com/viewLog.html?buildId=1688037&tab=artifacts#/cdc/tpcc-1000/rangefeed=true)
```
make stressrace TESTS=cdc/tpcc-1000/rangefeed=true PKG=./pkg/roachtest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```

<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).cdc/tpcc-1000/rangefeed=true failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1693614&tab=buildLog) on [master@e2c2be6cbef07c7f2ee6341816f81f72a206c32a](https://github.com/cockroachdb/cockroach/commits/e2c2be6cbef07c7f2ee6341816f81f72a206c32a):

```
		    -- arg 3: <string>
		  - /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1693614-1579160767-25-n4cpu16:4 -- ./workload run tpcc --warehouses=1000 --duration=120m --tolerate-errors {pgurl:1-3}  returned:
		    stderr:
		    
		    stdout:
		    atus
		       22.0s        0            0.0          258.8      0.0      0.0      0.0      0.0 payment
		       22.0s        0            7.0           25.8   3221.2   3489.7   3489.7   3489.7 stockLevel
		       23.0s        0           18.1           25.1   2818.6   3758.1   4026.5   4026.5 delivery
		       23.0s        0          207.3          116.0   4831.8   5100.3   5100.3   5100.3 newOrder
		       23.0s        0           16.1           27.2   3087.0   4026.5   4160.7   4160.7 orderStatus
		       23.0s        0           15.1          248.2   4295.0   4831.8   4831.8   4831.8 payment
		       23.0s        0           12.1           25.2   2952.8   4295.0   4295.0   4295.0 stockLevel
		       24.0s        0           47.9           26.1   2415.9   5637.1   5637.1   5637.1 delivery
		       24.0s        0          385.1          127.3   5905.6   6174.0   6174.0   6174.0 newOrder
		       24.0s        0           36.9           27.6   2550.1   4831.8   5368.7   5368.7 orderStatus
		       24.0s        0          188.6          245.7   4563.4   5637.1   5905.6   5905.6 payment
		       24.0s        0           25.9           25.2   2550.1   5100.3   5100.3   5100.3 stockLevel
		    _elapsed___errors__ops/sec(inst)___ops/sec(cum)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)
		       25.0s        0           20.0           25.8    570.4   3221.2   4831.8   4831.8 delivery
		       25.0s        0          844.2          156.0   6442.5   6979.3   6979.3   6979.3 newOrder
		       25.0s        0           36.9           28.0    536.9   2818.6   4563.4   4563.4 orderStatus
		       25.0s        0          235.5          245.3   3489.7   6174.0   6442.5   6710.9 payment
		       25.0s        0           20.0           25.0    838.9   4160.7   4563.4   4563.4 stockLevel
		       26.0s        0            4.0           25.0     44.0     56.6     56.6     56.6 delivery
		       26.0s        0            0.0          150.0      0.0      0.0      0.0      0.0 newOrder
		       26.0s        0            5.0           27.1      6.0      6.3      6.3      6.3 orderStatus
		       26.0s        0            2.0          236.0   2013.3   4831.8   4831.8   4831.8 payment
		       26.0s        0            0.0           24.0      0.0      0.0      0.0      0.0 stockLevel
		    E200116 08:50:55.786489 1 workload/cli/run.go:444  error in payment: EOF
		       27.0s      601            0.0           24.1      0.0      0.0      0.0      0.0 delivery
		       27.0s      601            2.0          144.6   2550.1   8321.5   8321.5   8321.5 newOrder
		       27.0s      601            0.0           26.1      0.0      0.0      0.0      0.0 orderStatus
		       27.0s      601            1.0          227.3   6710.9   6710.9   6710.9   6710.9 payment
		       27.0s      601            0.0           23.1      0.0      0.0      0.0      0.0 stockLevel
		    E200116 08:50:56.790161 1 workload/cli/run.go:444  error in newOrder: dial tcp 10.128.0.26:26257: connect: connection refused
		       28.0s      743            0.0           23.2      0.0      0.0      0.0      0.0 delivery
		       28.0s      743            0.0          139.4      0.0      0.0      0.0      0.0 newOrder
		       28.0s      743            1.0           25.2     88.1     88.1     88.1     88.1 orderStatus
		       28.0s      743            0.0          219.1      0.0      0.0      0.0      0.0 payment
		       28.0s      743            0.0           22.3      0.0      0.0      0.0      0.0 stockLevel
		    E200116 08:50:57.790162 1 workload/cli/run.go:444  error in orderStatus: dial tcp 10.128.0.26:26257: connect: connection refused
		    _elapsed___errors__ops/sec(inst)___ops/sec(cum)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)
		       29.0s      967            2.0           22.5    872.4   1006.6   1006.6   1006.6 delivery
		       29.0s      967            0.0          134.6      0.0      0.0      0.0      0.0 newOrder
		       29.0s      967            0.0           24.3      0.0      0.0      0.0      0.0 orderStatus
		       29.0s      967            0.0          211.6      0.0      0.0      0.0      0.0 payment
		       29.0s      967            0.0           21.6      0.0      0.0      0.0      0.0 stockLevel
		    E200116 08:50:58.790253 1 workload/cli/run.go:444  error in payment: dial tcp 10.128.0.26:26257: connect: connection refused:
		  - context canceled
```

<details><summary>Repro</summary><p>


Artifacts: [/cdc/tpcc-1000/rangefeed=true](https://teamcity.cockroachdb.com/viewLog.html?buildId=1693614&tab=artifacts#/cdc/tpcc-1000/rangefeed=true)
```
make stressrace TESTS=cdc/tpcc-1000/rangefeed=true PKG=./pkg/roachtest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```

<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).cdc/tpcc-1000/rangefeed=true failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1725352&tab=buildLog) on [master@4ec5929a08687b82b05ef80231fddd0d53421973](https://github.com/cockroachdb/cockroach/commits/4ec5929a08687b82b05ef80231fddd0d53421973):

```
		      429.0s        0           26.0           20.5     35.7     71.3     88.1     88.1 stockLevel
		      430.0s        0           29.0           20.6     60.8     88.1    109.1    109.1 delivery
		      430.0s        0          228.3          206.8     37.7     67.1     79.7     88.1 newOrder
		      430.0s        0           18.0           20.9      8.1     14.2     14.7     14.7 orderStatus
		      430.0s        0          216.3          209.9     24.1     46.1     54.5     54.5 payment
		      430.0s        0           29.0           20.5     50.3     71.3     79.7     79.7 stockLevel
		      431.0s        0           22.0           20.6     56.6     67.1     71.3     71.3 delivery
		      431.0s        0          225.0          206.9     33.6     48.2     71.3     79.7 newOrder
		      431.0s        0           23.0           20.9      8.1     11.5     14.7     14.7 orderStatus
		      431.0s        0          173.0          209.8     19.9     28.3     35.7     37.7 payment
		      431.0s        0           35.0           20.6     44.0     60.8     71.3     71.3 stockLevel
		      432.0s        0           23.0           20.7     60.8    109.1    113.2    113.2 delivery
		      432.0s        0          198.0          206.8     35.7     79.7     96.5    117.4 newOrder
		      432.0s        0           24.0           20.9      9.4     17.8     18.9     18.9 orderStatus
		      432.0s        0          242.0          209.9     21.0     48.2     58.7     65.0 payment
		      432.0s        0           30.0           20.6     46.1     92.3    100.7    100.7 stockLevel
		    _elapsed___errors__ops/sec(inst)___ops/sec(cum)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)
		      433.0s        0           25.0           20.7     52.4     62.9     62.9     62.9 delivery
		      433.0s        0          189.0          206.8     31.5     39.8     44.0     50.3 newOrder
		      433.0s        0           22.0           20.9      6.6     10.5     11.0     11.0 orderStatus
		      433.0s        0          217.0          209.9     16.3     22.0     25.2     32.5 payment
		      433.0s        0           19.0           20.6     39.8     52.4     75.5     75.5 stockLevel
		      434.0s        0           20.0           20.7     56.6    130.0    159.4    159.4 delivery
		      434.0s        0          224.6          206.8     35.7     75.5     96.5    100.7 newOrder
		      434.0s        0           16.0           20.9      6.6     16.3     22.0     22.0 orderStatus
		      434.0s        0          220.6          209.9     21.0     39.8     65.0     75.5 payment
		      434.0s        0           25.0           20.6     46.1     79.7     79.7     79.7 stockLevel:
		  - context canceled

	cluster.go:2368,cdc.go:224,cdc.go:521,test_runner.go:741: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2356
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2364
		    main.cdcBasicTest
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:224
		    main.registerCDC.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:521
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:741
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - error with attached stack trace:
		    main.(*monitor).wait.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2412
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor task failed:
		  - context canceled

```

<details><summary>More</summary><p>


Artifacts: [/cdc/tpcc-1000/rangefeed=true](https://teamcity.cockroachdb.com/viewLog.html?buildId=1725352&tab=artifacts#/cdc/tpcc-1000/rangefeed=true)
Related:
- #44572 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202001302015_v19.2.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202001302015_v19.2.3) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Acdc%2Ftpcc-1000%2Frangefeed%3Dtrue.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).cdc/tpcc-1000/rangefeed=true failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1737376&tab=buildLog) on [master@08bb94d98b7f60126666e9c5a317afca61422d7f](https://github.com/cockroachdb/cockroach/commits/08bb94d98b7f60126666e9c5a317afca61422d7f):

```
		     1045.0s        0            7.0           20.9   4295.0   6710.9   6710.9   6710.9 stockLevel
		     1046.0s        0            0.0           20.9      0.0      0.0      0.0      0.0 delivery
		     1046.0s        0            0.0          206.7      0.0      0.0      0.0      0.0 newOrder
		     1046.0s        0            0.0           21.0      0.0      0.0      0.0      0.0 orderStatus
		     1046.0s        0            0.0          208.7      0.0      0.0      0.0      0.0 payment
		     1046.0s        0            0.0           20.9      0.0      0.0      0.0      0.0 stockLevel
		     1047.0s     1002            0.0           20.9      0.0      0.0      0.0      0.0 delivery
		     1047.0s     1002            1.0          206.5   9126.8   9126.8   9126.8   9126.8 newOrder
		     1047.0s     1002            1.0           20.9   4563.4   4563.4   4563.4   4563.4 orderStatus
		     1047.0s     1002            0.0          208.5      0.0      0.0      0.0      0.0 payment
		     1047.0s     1002            0.0           20.9      0.0      0.0      0.0      0.0 stockLevel
		     1048.0s     1448            0.0           20.8      0.0      0.0      0.0      0.0 delivery
		     1048.0s     1448            0.0          206.4      0.0      0.0      0.0      0.0 newOrder
		     1048.0s     1448            0.0           20.9      0.0      0.0      0.0      0.0 orderStatus
		     1048.0s     1448            0.0          208.3      0.0      0.0      0.0      0.0 payment
		     1048.0s     1448            0.0           20.9      0.0      0.0      0.0      0.0 stockLevel
		    _elapsed___errors__ops/sec(inst)___ops/sec(cum)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)
		     1049.0s     1671            0.0           20.8      0.0      0.0      0.0      0.0 delivery
		     1049.0s     1671            0.0          206.2      0.0      0.0      0.0      0.0 newOrder
		     1049.0s     1671            0.0           20.9      0.0      0.0      0.0      0.0 orderStatus
		     1049.0s     1671            0.0          208.1      0.0      0.0      0.0      0.0 payment
		     1049.0s     1671            0.0           20.9      0.0      0.0      0.0      0.0 stockLevel
		     1050.0s     2282            0.0           20.8      0.0      0.0      0.0      0.0 delivery
		     1050.0s     2282            0.0          206.0      0.0      0.0      0.0      0.0 newOrder
		     1050.0s     2282            0.0           20.9      0.0      0.0      0.0      0.0 orderStatus
		     1050.0s     2282            0.0          207.9      0.0      0.0      0.0      0.0 payment
		     1050.0s     2282            0.0           20.8      0.0      0.0      0.0      0.0 stockLevel:
		  - context canceled

	cluster.go:2368,cdc.go:224,cdc.go:521,test_runner.go:741: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2356
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2364
		    main.cdcBasicTest
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:224
		    main.registerCDC.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:521
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:741
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - unexpected node event: 2: dead

	cluster.go:1410,context.go:135,cluster.go:1399,test_runner.go:778: dead node detection: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod monitor teamcity-1737376-1581149318-25-n4cpu16 --oneshot --ignore-empty-nodes: exit status 1 4: skipped
		2: dead
		3: 4327
		1: 4732
		Error:  2: dead

```

<details><summary>More</summary><p>


Artifacts: [/cdc/tpcc-1000/rangefeed=true](https://teamcity.cockroachdb.com/viewLog.html?buildId=1737376&tab=artifacts#/cdc/tpcc-1000/rangefeed=true)
Related:
- #44572 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202001302015_v19.2.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202001302015_v19.2.3) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Acdc%2Ftpcc-1000%2Frangefeed%3Dtrue.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).cdc/tpcc-1000/rangefeed=true failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1793930&tab=buildLog) on [master@752dea867f3aeb142e98c22f8d320ce19041aa8d](https://github.com/cockroachdb/cockroach/commits/752dea867f3aeb142e98c22f8d320ce19041aa8d):

```
		       51.0s        0           24.0           23.7     54.5    100.7    117.4    117.4 stockLevel
		       52.0s      122           12.0           21.9     56.6     65.0     71.3     71.3 delivery
		       52.0s      122           66.0          172.9     32.5   4160.7   9126.8  10737.4 newOrder
		       52.0s      122            3.0           22.8      7.3      9.4      9.4      9.4 orderStatus
		       52.0s      122           53.0          229.0     17.8     25.2     27.3     31.5 payment
		       52.0s      122            9.0           23.4     58.7    335.5    335.5    335.5 stockLevel
		    _elapsed___errors__ops/sec(inst)___ops/sec(cum)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)
		       53.0s      287            2.0           21.5     46.1     50.3     50.3     50.3 delivery
		       53.0s      287            0.0          169.6      0.0      0.0      0.0      0.0 newOrder
		       53.0s      287            0.0           22.4      0.0      0.0      0.0      0.0 orderStatus
		       53.0s      287            0.0          224.6      0.0      0.0      0.0      0.0 payment
		       53.0s      287            5.0           23.1     52.4     75.5     75.5     75.5 stockLevel
		       54.0s      479            0.0           21.1      0.0      0.0      0.0      0.0 delivery
		       54.0s      479            0.0          166.5      0.0      0.0      0.0      0.0 newOrder
		       54.0s      479            1.0           22.0      5.2      5.2      5.2      5.2 orderStatus
		       54.0s      479            0.0          220.5      0.0      0.0      0.0      0.0 payment
		       54.0s      479            7.0           22.8     60.8     71.3     71.3     71.3 stockLevel
		       55.0s      737            0.0           20.7      0.0      0.0      0.0      0.0 delivery
		       55.0s      737            0.0          163.4      0.0      0.0      0.0      0.0 newOrder
		       55.0s      737            0.0           21.6      0.0      0.0      0.0      0.0 orderStatus
		       55.0s      737            0.0          216.5      0.0      0.0      0.0      0.0 payment
		       55.0s      737            5.0           22.5     54.5     83.9     83.9     83.9 stockLevel
		       56.0s      959            0.0           20.4      0.0      0.0      0.0      0.0 delivery
		       56.0s      959            0.0          160.5      0.0      0.0      0.0      0.0 newOrder
		       56.0s      959            0.0           21.2      0.0      0.0      0.0      0.0 orderStatus
		       56.0s      959            0.0          212.6      0.0      0.0      0.0      0.0 payment
		       56.0s      959            3.0           22.1     26.2     58.7     58.7     58.7 stockLevel:
		  - context canceled

	cluster.go:2368,cdc.go:224,cdc.go:521,test_runner.go:741: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2356
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2364
		    main.cdcBasicTest
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:224
		    main.registerCDC.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:521
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:741
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - unexpected node event: 2: dead

	cluster.go:1410,context.go:135,cluster.go:1399,test_runner.go:778: dead node detection: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod monitor teamcity-1793930-1583568484-25-n4cpu16 --oneshot --ignore-empty-nodes: exit status 1 4: skipped
		2: dead
		1: 4931
		3: 4184
		Error:  2: dead

```

<details><summary>More</summary><p>


Artifacts: [/cdc/tpcc-1000/rangefeed=true](https://teamcity.cockroachdb.com/viewLog.html?buildId=1793930&tab=artifacts#/cdc/tpcc-1000/rangefeed=true)
Related:
- #45437 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44971 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.1) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44572 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202001302015_v19.2.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202001302015_v19.2.3) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Acdc%2Ftpcc-1000%2Frangefeed%3Dtrue.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).cdc/tpcc-1000/rangefeed=true failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1795062&tab=buildLog) on [master@dfa5bd527ae7d7373dd03c62118df87a87a77130](https://github.com/cockroachdb/cockroach/commits/dfa5bd527ae7d7373dd03c62118df87a87a77130):

```
		       25.0s        0          415.1          107.5   6442.5   6442.5   6442.5   6442.5 newOrder
		       25.0s        0            3.0           21.7   1208.0   4295.0   4295.0   4295.0 orderStatus
		       25.0s        0           30.0          190.6  17179.9  21474.8  21474.8  21474.8 payment
		       25.0s        0            6.0           20.6   2281.7   4295.0   4295.0   4295.0 stockLevel
		       26.0s     1270            0.0           20.7      0.0      0.0      0.0      0.0 delivery
		       26.0s     1270          150.2          109.1   7784.6   7784.6   7784.6   7784.6 newOrder
		       26.0s     1270            1.0           20.9   2550.1   2550.1   2550.1   2550.1 orderStatus
		       26.0s     1270            0.0          183.3      0.0      0.0      0.0      0.0 payment
		       26.0s     1270            4.0           20.0   2281.7   3221.2   3221.2   3221.2 stockLevel
		       27.0s     1386            0.0           20.0      0.0      0.0      0.0      0.0 delivery
		       27.0s     1386            0.0          105.1      0.0      0.0      0.0      0.0 newOrder
		       27.0s     1386            0.0           20.1      0.0      0.0      0.0      0.0 orderStatus
		       27.0s     1386            0.0          176.5      0.0      0.0      0.0      0.0 payment
		       27.0s     1386            0.0           19.3      0.0      0.0      0.0      0.0 stockLevel
		       28.0s     1652            0.0           19.2      0.0      0.0      0.0      0.0 delivery
		       28.0s     1652            0.0          101.3      0.0      0.0      0.0      0.0 newOrder
		       28.0s     1652            0.0           19.4      0.0      0.0      0.0      0.0 orderStatus
		       28.0s     1652            0.0          170.2      0.0      0.0      0.0      0.0 payment
		       28.0s     1652            0.0           18.6      0.0      0.0      0.0      0.0 stockLevel:
		  - context canceled

	cluster.go:2368,cdc.go:224,cdc.go:521,test_runner.go:741: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2356
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2364
		    main.cdcBasicTest
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:224
		    main.registerCDC.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:521
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:741
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - error with attached stack trace:
		    main.(*monitor).wait.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2412
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor task failed:
		  - dial tcp 35.239.224.89:26257
		  - connect
		  - connection refused

	cluster.go:1410,context.go:135,cluster.go:1399,test_runner.go:778: dead node detection: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod monitor teamcity-1795062-1583655253-25-n4cpu16 --oneshot --ignore-empty-nodes: exit status 1 4: skipped
		1: dead
		2: 4104
		3: 4321
		Error:  1: dead

```

<details><summary>More</summary><p>


Artifacts: [/cdc/tpcc-1000/rangefeed=true](https://teamcity.cockroachdb.com/viewLog.html?buildId=1795062&tab=artifacts#/cdc/tpcc-1000/rangefeed=true)
Related:
- #45437 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44971 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.1) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44572 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202001302015_v19.2.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202001302015_v19.2.3) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Acdc%2Ftpcc-1000%2Frangefeed%3Dtrue.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).cdc/tpcc-1000/rangefeed=true failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1796240&tab=buildLog) on [master@c473f40078994551cebcbe00fdbf1fa388957658](https://github.com/cockroachdb/cockroach/commits/c473f40078994551cebcbe00fdbf1fa388957658):

```
		      417.0s        0           17.0           20.7     60.8    142.6    159.4    159.4 stockLevel
		      418.0s        0           22.0           20.5     56.6     83.9    104.9    104.9 delivery
		      418.0s        0          202.0          206.4     35.7     67.1   9126.8  10737.4 newOrder
		      418.0s        0           17.0           20.5      8.4     14.7     28.3     28.3 orderStatus
		      418.0s        0          206.0          208.3     17.8     32.5    520.1  10737.4 payment
		      418.0s        0           15.0           20.7     56.6    121.6    125.8    125.8 stockLevel
		      419.0s        0           20.0           20.5     65.0     96.5     96.5     96.5 delivery
		      419.0s        0          159.1          206.3     39.8     67.1     88.1  10737.4 newOrder
		      419.0s        0           18.0           20.5      7.9     13.6     15.2     15.2 orderStatus
		      419.0s        0          145.1          208.2     19.9     33.6     39.8     39.8 payment
		      419.0s        0           18.0           20.7     71.3    113.2    159.4    159.4 stockLevel
		      420.0s      231            2.0           20.5     35.7     44.0     44.0     44.0 delivery
		      420.0s      231            0.0          205.8      0.0      0.0      0.0      0.0 newOrder
		      420.0s      231            3.0           20.5      6.8     88.1     88.1     88.1 orderStatus
		      420.0s      231            5.0          207.7     15.7     21.0     21.0     21.0 payment
		      420.0s      231            4.0           20.6     25.2    285.2    285.2    285.2 stockLevel
		    _elapsed___errors__ops/sec(inst)___ops/sec(cum)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)
		      421.0s      378            4.0           20.4     48.2     52.4     52.4     52.4 delivery
		      421.0s      378            2.0          205.4      6.3      7.1      7.1      7.1 newOrder
		      421.0s      378            4.0           20.4      7.3     14.7     14.7     14.7 orderStatus
		      421.0s      378           11.0          207.2     12.6     16.3     18.9     18.9 payment
		      421.0s      378            5.0           20.6     46.1     79.7     79.7     79.7 stockLevel
		      422.0s      582            1.0           20.4     44.0     44.0     44.0     44.0 delivery
		      422.0s      582            0.0          204.9      0.0      0.0      0.0      0.0 newOrder
		      422.0s      582            2.0           20.4      5.5      6.6      6.6      6.6 orderStatus
		      422.0s      582            7.0          206.7     14.2     16.3     16.3     16.3 payment
		      422.0s      582            1.0           20.6     44.0     44.0     44.0     44.0 stockLevel:
		  - context canceled

	cluster.go:2368,cdc.go:224,cdc.go:521,test_runner.go:741: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2356
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2364
		    main.cdcBasicTest
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:224
		    main.registerCDC.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:521
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:741
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - unexpected node event: 3: dead

	cluster.go:1410,context.go:135,cluster.go:1399,test_runner.go:778: dead node detection: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod monitor teamcity-1796240-1583738442-26-n4cpu16 --oneshot --ignore-empty-nodes: exit status 1 4: skipped
		3: dead
		2: 4597
		1: 4786
		Error:  3: dead

```

<details><summary>More</summary><p>


Artifacts: [/cdc/tpcc-1000/rangefeed=true](https://teamcity.cockroachdb.com/viewLog.html?buildId=1796240&tab=artifacts#/cdc/tpcc-1000/rangefeed=true)
Related:
- #45437 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44971 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.1) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44572 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202001302015_v19.2.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202001302015_v19.2.3) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Acdc%2Ftpcc-1000%2Frangefeed%3Dtrue.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).cdc/tpcc-1000/rangefeed=true failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1804460&tab=buildLog) on [master@793a9200c16693aff32aa6a4dd9d8bbcbddb30aa](https://github.com/cockroachdb/cockroach/commits/793a9200c16693aff32aa6a4dd9d8bbcbddb30aa):

```
		       33.0s        0           30.0           24.5     41.9     79.7     92.3     92.3 stockLevel
		       34.0s        0           26.0           24.3     62.9    192.9    201.3    201.3 delivery
		       34.0s        0          152.0          161.4     48.2    151.0    167.8    184.5 newOrder
		       34.0s        0           20.0           23.6      7.6     24.1     24.1     24.1 orderStatus
		       34.0s        0          222.0          240.9     23.1     96.5    142.6    176.2 payment
		       34.0s        0           32.0           24.7     46.1     92.3    134.2    134.2 stockLevel
		       35.0s        0           18.0           24.1     48.2     58.7     83.9     83.9 delivery
		       35.0s        0          105.0          159.8     33.6   2281.7  12884.9  16643.0 newOrder
		       35.0s        0           11.0           23.3      6.6      9.4     35.7     35.7 orderStatus
		       35.0s        0          118.0          237.4     18.9     41.9     54.5  13421.8 payment
		       35.0s        0           11.0           24.3     37.7     44.0     58.7     58.7 stockLevel
		       36.0s      259            0.0           23.4      0.0      0.0      0.0      0.0 delivery
		       36.0s      259            0.0          155.4      0.0      0.0      0.0      0.0 newOrder
		       36.0s      259            5.0           22.8      6.6      7.9      7.9      7.9 orderStatus
		       36.0s      259            0.0          230.8      0.0      0.0      0.0      0.0 payment
		       36.0s      259            0.0           23.7      0.0      0.0      0.0      0.0 stockLevel
		    _elapsed___errors__ops/sec(inst)___ops/sec(cum)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)
		       37.0s      412            2.0           22.9     39.8     52.4     52.4     52.4 delivery
		       37.0s      412            0.0          151.2      0.0      0.0      0.0      0.0 newOrder
		       37.0s      412            5.0           22.3      7.1      9.4      9.4      9.4 orderStatus
		       37.0s      412            0.0          224.5      0.0      0.0      0.0      0.0 payment
		       37.0s      412            0.0           23.0      0.0      0.0      0.0      0.0 stockLevel
		       38.0s      584            1.0           22.3     33.6     33.6     33.6     33.6 delivery
		       38.0s      584            0.0          147.2      0.0      0.0      0.0      0.0 newOrder
		       38.0s      584            2.0           21.8      6.0      6.6      6.6      6.6 orderStatus
		       38.0s      584            0.0          218.6      0.0      0.0      0.0      0.0 payment
		       38.0s      584            0.0           22.4      0.0      0.0      0.0      0.0 stockLevel:
		  - context canceled

	cluster.go:2368,cdc.go:224,cdc.go:521,test_runner.go:747: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2356
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2364
		    main.cdcBasicTest
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:224
		    main.registerCDC.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:521
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:747
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - unexpected node event: 2: dead

	cluster.go:1410,context.go:135,cluster.go:1399,test_runner.go:801: dead node detection: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod monitor teamcity-1804460-1583998462-27-n4cpu16 --oneshot --ignore-empty-nodes: exit status 1 4: skipped
		2: dead
		3: 4764
		1: 4234
		Error:  2: dead

```

<details><summary>More</summary><p>


Artifacts: [/cdc/tpcc-1000/rangefeed=true](https://teamcity.cockroachdb.com/viewLog.html?buildId=1804460&tab=artifacts#/cdc/tpcc-1000/rangefeed=true)
Related:
- #45437 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44971 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.1) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44572 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202001302015_v19.2.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202001302015_v19.2.3) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Acdc%2Ftpcc-1000%2Frangefeed%3Dtrue.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).cdc/tpcc-1000/rangefeed=true failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1806460&tab=buildLog) on [master@69dc87d68addedf2fabfb2b14c098cfb35b5f3d0](https://github.com/cockroachdb/cockroach/commits/69dc87d68addedf2fabfb2b14c098cfb35b5f3d0):

```
		     1443.0s        0           25.6           21.0     67.1    121.6    142.6    142.6 stockLevel
		     1444.0s        0           15.0           20.9     79.7    104.9    113.2    113.2 delivery
		     1444.0s        0          215.6          208.5     46.1     92.3    113.2    117.4 newOrder
		     1444.0s        0           24.0           20.9      8.9     14.2     15.2     15.2 orderStatus
		     1444.0s        0          220.6          209.8     26.2     54.5     75.5     79.7 payment
		     1444.0s        0           18.0           21.0     65.0    109.1    109.1    109.1 stockLevel
		    _elapsed___errors__ops/sec(inst)___ops/sec(cum)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)
		     1445.0s        0            5.0           20.9     83.9    109.1    109.1    109.1 delivery
		     1445.0s        0           87.1          208.5     50.3     75.5   4831.8  10737.4 newOrder
		     1445.0s        0           12.0           20.9      7.9     14.2     15.7     15.7 orderStatus
		     1445.0s        0           72.1          209.7     28.3     37.7     39.8     39.8 payment
		     1445.0s        0           10.0           21.0     54.5    109.1    109.1    109.1 stockLevel
		     1446.0s      268            1.0           20.9     41.9     41.9     41.9     41.9 delivery
		     1446.0s      268            0.0          208.3      0.0      0.0      0.0      0.0 newOrder
		     1446.0s      268            1.0           20.9      9.4      9.4      9.4      9.4 orderStatus
		     1446.0s      268            0.0          209.6      0.0      0.0      0.0      0.0 payment
		     1446.0s      268            3.0           21.0     54.5     79.7     79.7     79.7 stockLevel
		     1447.0s      433            0.0           20.9      0.0      0.0      0.0      0.0 delivery
		     1447.0s      433            0.0          208.2      0.0      0.0      0.0      0.0 newOrder
		     1447.0s      433            1.0           20.9      8.1      8.1      8.1      8.1 orderStatus
		     1447.0s      433            0.0          209.4      0.0      0.0      0.0      0.0 payment
		     1447.0s      433            1.0           21.0     33.6     33.6     33.6     33.6 stockLevel
		     1448.0s      619            0.0           20.9      0.0      0.0      0.0      0.0 delivery
		     1448.0s      619            0.0          208.0      0.0      0.0      0.0      0.0 newOrder
		     1448.0s      619            2.0           20.9      5.0     11.0     11.0     11.0 orderStatus
		     1448.0s      619            0.0          209.3      0.0      0.0      0.0      0.0 payment
		     1448.0s      619            2.0           20.9     44.0     58.7     58.7     58.7 stockLevel:
		  - context canceled

	cluster.go:2368,cdc.go:224,cdc.go:521,test_runner.go:753: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2356
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2364
		    main.cdcBasicTest
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:224
		    main.registerCDC.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:521
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:753
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - unexpected node event: 2: dead

	cluster.go:1410,context.go:135,cluster.go:1399,test_runner.go:807: dead node detection: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod monitor teamcity-1806460-1584083876-26-n4cpu16 --oneshot --ignore-empty-nodes: exit status 1 4: skipped
		2: dead
		3: 4174
		1: 4415
		Error:  2: dead

```

<details><summary>More</summary><p>


Artifacts: [/cdc/tpcc-1000/rangefeed=true](https://teamcity.cockroachdb.com/viewLog.html?buildId=1806460&tab=artifacts#/cdc/tpcc-1000/rangefeed=true)
Related:
- #45437 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44971 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.1) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44572 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202001302015_v19.2.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202001302015_v19.2.3) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Acdc%2Ftpcc-1000%2Frangefeed%3Dtrue.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).cdc/tpcc-1000/rangefeed=true failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1808695&tab=buildLog) on [master@33d71472dc01cbc5064b3c5e1fcd666a33f606de](https://github.com/cockroachdb/cockroach/commits/33d71472dc01cbc5064b3c5e1fcd666a33f606de):

```
		      441.0s        0          126.2          207.1     37.7     56.6  10200.5  10737.4 newOrder
		      441.0s        0           10.0           20.9      7.3     12.6     12.6     12.6 orderStatus
		      441.0s        0          120.2          209.5     24.1     35.7     41.9     44.0 payment
		      441.0s        0           12.0           20.7     48.2     71.3     71.3     71.3 stockLevel
		      442.0s      253            1.0           20.7     44.0     44.0     44.0     44.0 delivery
		      442.0s      253            0.0          206.6      0.0      0.0      0.0      0.0 newOrder
		      442.0s      253            4.0           20.9      7.9    369.1    369.1    369.1 orderStatus
		      442.0s      253            0.0          209.0      0.0      0.0      0.0      0.0 payment
		      442.0s      253            4.0           20.6     29.4     65.0     65.0     65.0 stockLevel
		      443.0s      419            2.0           20.7     50.3     58.7     58.7     58.7 delivery
		      443.0s      419            0.0          206.1      0.0      0.0      0.0      0.0 newOrder
		      443.0s      419            0.0           20.8      0.0      0.0      0.0      0.0 orderStatus
		      443.0s      419            0.0          208.5      0.0      0.0      0.0      0.0 payment
		      443.0s      419            7.0           20.6     44.0     48.2     48.2     48.2 stockLevel
		      444.0s      623            0.0           20.6      0.0      0.0      0.0      0.0 delivery
		      444.0s      623            0.0          205.7      0.0      0.0      0.0      0.0 newOrder
		      444.0s      623            1.0           20.8      5.2      5.2      5.2      5.2 orderStatus
		      444.0s      623            0.0          208.0      0.0      0.0      0.0      0.0 payment
		      444.0s      623            5.0           20.6     30.4     44.0     44.0     44.0 stockLevel:
		  - context canceled

	cluster.go:2368,cdc.go:224,cdc.go:521,test_runner.go:753: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2356
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2364
		    main.cdcBasicTest
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:224
		    main.registerCDC.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:521
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:753
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - error with attached stack trace:
		    main.(*monitor).wait.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2412
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor task failed:
		  - dial tcp 34.67.129.223:26257
		  - connect
		  - connection refused

	cluster.go:1410,context.go:135,cluster.go:1399,test_runner.go:807: dead node detection: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod monitor teamcity-1808695-1584256913-27-n4cpu16 --oneshot --ignore-empty-nodes: exit status 1 4: skipped
		1: dead
		3: 4457
		2: 4584
		Error:  1: dead

```

<details><summary>More</summary><p>


Artifacts: [/cdc/tpcc-1000/rangefeed=true](https://teamcity.cockroachdb.com/viewLog.html?buildId=1808695&tab=artifacts#/cdc/tpcc-1000/rangefeed=true)
Related:
- #45437 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44971 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.1) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44572 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202001302015_v19.2.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202001302015_v19.2.3) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Acdc%2Ftpcc-1000%2Frangefeed%3Dtrue.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).cdc/tpcc-1000/rangefeed=true failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1809758&tab=buildLog) on [master@d8bdc938177ea4b5226b95ee6cc51af0951054fc](https://github.com/cockroachdb/cockroach/commits/d8bdc938177ea4b5226b95ee6cc51af0951054fc):

```
		      290.0s        0           15.0           18.3     54.5    125.8    134.2    134.2 stockLevel
		      291.0s        0           12.0           18.4     54.5     71.3    100.7    100.7 delivery
		      291.0s        0          200.2          182.2     35.7     46.1     60.8     62.9 newOrder
		      291.0s        0           16.0           18.7      7.3     12.1     14.2     14.2 orderStatus
		      291.0s        0          223.2          186.4     19.9     28.3     35.7     37.7 payment
		      291.0s        0           16.0           18.3     46.1     83.9     96.5     96.5 stockLevel
		      292.0s        0           10.0           18.4     62.9     71.3     71.3     71.3 delivery
		      292.0s        0          141.8          182.0     37.7     52.4   8589.9  10737.4 newOrder
		      292.0s        0           17.0           18.6      7.3     10.0     12.1     12.1 orderStatus
		      292.0s        0          135.9          186.2     18.9     26.2     29.4     31.5 payment
		      292.0s        0           15.0           18.3     44.0     56.6     60.8     60.8 stockLevel
		    _elapsed___errors__ops/sec(inst)___ops/sec(cum)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)
		      293.0s      223            0.0           18.3      0.0      0.0      0.0      0.0 delivery
		      293.0s      223            0.0          181.4      0.0      0.0      0.0      0.0 newOrder
		      293.0s      223            3.0           18.6      8.1     13.6     13.6     13.6 orderStatus
		      293.0s      223           11.0          185.6     13.1     18.9     22.0     22.0 payment
		      293.0s      223            5.0           18.2     28.3     54.5     54.5     54.5 stockLevel
		      294.0s      376            0.0           18.2      0.0      0.0      0.0      0.0 delivery
		      294.0s      376            0.0          180.8      0.0      0.0      0.0      0.0 newOrder
		      294.0s      376            2.0           18.5      6.0      8.9      8.9      8.9 orderStatus
		      294.0s      376            7.0          185.0     12.6     15.7     15.7     15.7 payment
		      294.0s      376            4.0           18.2     35.7     60.8     60.8     60.8 stockLevel
		      295.0s      554            0.0           18.2      0.0      0.0      0.0      0.0 delivery
		      295.0s      554            0.0          180.2      0.0      0.0      0.0      0.0 newOrder
		      295.0s      554            2.0           18.5      7.3      7.6      7.6      7.6 orderStatus
		      295.0s      554            6.0          184.4     11.5     13.6     13.6     13.6 payment
		      295.0s      554            3.0           18.1     35.7     50.3     50.3     50.3 stockLevel:
		  - context canceled

	cluster.go:2368,cdc.go:224,cdc.go:521,test_runner.go:753: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2356
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2364
		    main.cdcBasicTest
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:224
		    main.registerCDC.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:521
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:753
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - unexpected node event: 3: dead

	cluster.go:1410,context.go:135,cluster.go:1399,test_runner.go:807: dead node detection: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod monitor teamcity-1809758-1584345030-27-n4cpu16 --oneshot --ignore-empty-nodes: exit status 1 4: skipped
		3: dead
		1: 4448
		2: 4535
		Error:  3: dead

```

<details><summary>More</summary><p>


Artifacts: [/cdc/tpcc-1000/rangefeed=true](https://teamcity.cockroachdb.com/viewLog.html?buildId=1809758&tab=artifacts#/cdc/tpcc-1000/rangefeed=true)
Related:
- #46137 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202003112118_v20.1.0-beta.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202003112118_v20.1.0-beta.3) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #45437 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44971 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.1) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44572 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202001302015_v19.2.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202001302015_v19.2.3) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Acdc%2Ftpcc-1000%2Frangefeed%3Dtrue.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).cdc/tpcc-1000/rangefeed=true failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1814553&tab=buildLog) on [master@b5f030223fbcf22e806c48a3c46e74a73a54f50f](https://github.com/cockroachdb/cockroach/commits/b5f030223fbcf22e806c48a3c46e74a73a54f50f):

```
		     2817.0s        0          126.0          208.7     37.7     54.5  10200.5  10737.4 newOrder
		     2817.0s        0           12.0           20.9      7.1     10.5     13.6     13.6 orderStatus
		     2817.0s        0           99.0          209.3     19.9     46.1     54.5     79.7 payment
		     2817.0s        0            9.0           20.9     39.8     54.5     54.5     54.5 stockLevel
		     2818.0s      252            0.0           20.9      0.0      0.0      0.0      0.0 delivery
		     2818.0s      252            0.0          208.7      0.0      0.0      0.0      0.0 newOrder
		     2818.0s      252            7.0           20.9      7.6     14.2     14.2     14.2 orderStatus
		     2818.0s      252            0.0          209.3      0.0      0.0      0.0      0.0 payment
		     2818.0s      252            5.0           20.9     37.7     52.4     52.4     52.4 stockLevel
		     2819.0s      394            3.0           20.9     48.2     56.6     56.6     56.6 delivery
		     2819.0s      394            0.0          208.6      0.0      0.0      0.0      0.0 newOrder
		     2819.0s      394            1.0           20.9      5.8      5.8      5.8      5.8 orderStatus
		     2819.0s      394            0.0          209.2      0.0      0.0      0.0      0.0 payment
		     2819.0s      394            7.0           20.9     41.9     46.1     46.1     46.1 stockLevel
		     2820.0s      599            0.0           20.9      0.0      0.0      0.0      0.0 delivery
		     2820.0s      599            0.0          208.5      0.0      0.0      0.0      0.0 newOrder
		     2820.0s      599            1.0           20.9      8.4      8.4      8.4      8.4 orderStatus
		     2820.0s      599            0.0          209.1      0.0      0.0      0.0      0.0 payment
		     2820.0s      599            3.0           20.9     41.9     52.4     52.4     52.4 stockLevel:
		  - context canceled

	cluster.go:2368,cdc.go:224,cdc.go:521,test_runner.go:753: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2356
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2364
		    main.cdcBasicTest
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:224
		    main.registerCDC.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:521
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:753
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - error with attached stack trace:
		    main.(*monitor).wait.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2412
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor task failed:
		  - read tcp 172.17.0.2:49246->35.239.33.247:26257
		  - read
		  - connection reset by peer

	cluster.go:1410,context.go:135,cluster.go:1399,test_runner.go:825: dead node detection: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod monitor teamcity-1814553-1584515366-25-n4cpu16 --oneshot --ignore-empty-nodes: exit status 1 4: skipped
		1: dead
		3: 4238
		2: 4313
		Error:  1: dead

```

<details><summary>More</summary><p>


Artifacts: [/cdc/tpcc-1000/rangefeed=true](https://teamcity.cockroachdb.com/viewLog.html?buildId=1814553&tab=artifacts#/cdc/tpcc-1000/rangefeed=true)
Related:
- #46137 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202003112118_v20.1.0-beta.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202003112118_v20.1.0-beta.3) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #45437 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44971 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.1) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44572 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202001302015_v19.2.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202001302015_v19.2.3) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Acdc%2Ftpcc-1000%2Frangefeed%3Dtrue.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).cdc/tpcc-1000/rangefeed=true failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1816964&tab=buildLog) on [master@d571624ae4f833c49c717728a74cc7be78a791f0](https://github.com/cockroachdb/cockroach/commits/d571624ae4f833c49c717728a74cc7be78a791f0):

```
		     1017.0s        0           22.0           21.0     54.5    121.6    167.8    167.8 stockLevel
		     1018.0s        0           17.0           21.0     79.7    109.1    121.6    121.6 delivery
		     1018.0s        0          222.1          208.5     44.0    113.2   3489.7  10737.4 newOrder
		     1018.0s        0           30.0           21.1      8.9     25.2     31.5     31.5 orderStatus
		     1018.0s        0          205.1          210.5     23.1     62.9     79.7     83.9 payment
		     1018.0s        0           24.0           21.0     44.0     71.3     96.5     96.5 stockLevel
		     1019.0s      170            0.0           21.0      0.0      0.0      0.0      0.0 delivery
		     1019.0s      170            2.0          208.3     35.7     41.9     41.9     41.9 newOrder
		     1019.0s      170            0.0           21.1      0.0      0.0      0.0      0.0 orderStatus
		     1019.0s      170            0.0          210.3      0.0      0.0      0.0      0.0 payment
		     1019.0s      170            5.0           21.0     35.7     58.7     58.7     58.7 stockLevel
		     1020.0s      329            1.0           21.0     50.3     50.3     50.3     50.3 delivery
		     1020.0s      329            0.0          208.1      0.0      0.0      0.0      0.0 newOrder
		     1020.0s      329            0.0           21.0      0.0      0.0      0.0      0.0 orderStatus
		     1020.0s      329            0.0          210.0      0.0      0.0      0.0      0.0 payment
		     1020.0s      329            2.0           21.0     18.9     25.2     25.2     25.2 stockLevel
		    _elapsed___errors__ops/sec(inst)___ops/sec(cum)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)
		     1021.0s      495            0.0           20.9      0.0      0.0      0.0      0.0 delivery
		     1021.0s      495            0.0          207.9      0.0      0.0      0.0      0.0 newOrder
		     1021.0s      495            0.0           21.0      0.0      0.0      0.0      0.0 orderStatus
		     1021.0s      495            0.0          209.8      0.0      0.0      0.0      0.0 payment
		     1021.0s      495            1.0           21.0    838.9    838.9    838.9    838.9 stockLevel
		     1022.0s      764            0.0           20.9      0.0      0.0      0.0      0.0 delivery
		     1022.0s      764            0.0          207.7      0.0      0.0      0.0      0.0 newOrder
		     1022.0s      764            0.0           21.0      0.0      0.0      0.0      0.0 orderStatus
		     1022.0s      764            0.0          209.6      0.0      0.0      0.0      0.0 payment
		     1022.0s      764            1.0           21.0   2080.4   2080.4   2080.4   2080.4 stockLevel:
		  - context canceled

	cluster.go:2368,cdc.go:224,cdc.go:521,test_runner.go:753: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2356
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2364
		    main.cdcBasicTest
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:224
		    main.registerCDC.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:521
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:753
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - unexpected node event: 3: dead

	cluster.go:1410,context.go:135,cluster.go:1399,test_runner.go:825: dead node detection: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod monitor teamcity-1816964-1584603715-24-n4cpu16 --oneshot --ignore-empty-nodes: exit status 1 4: skipped
		3: dead
		2: 4584
		1: 4505
		Error:  3: dead

```

<details><summary>More</summary><p>


Artifacts: [/cdc/tpcc-1000/rangefeed=true](https://teamcity.cockroachdb.com/viewLog.html?buildId=1816964&tab=artifacts#/cdc/tpcc-1000/rangefeed=true)
Related:
- #46290 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202003181957_v20.1.0-beta.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202003181957_v20.1.0-beta.3) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #46137 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202003112118_v20.1.0-beta.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202003112118_v20.1.0-beta.3) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #45437 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44971 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.1) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44572 roachtest: cdc/tpcc-1000/rangefeed=true failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202001302015_v19.2.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202001302015_v19.2.3) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Acdc%2Ftpcc-1000%2Frangefeed%3Dtrue.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/kv/kvserver/concurrency/concurrency_manager_test.go
+++ b/pkg/kv/kvserver/concurrency/concurrency_manager_test.go
@@ -53,15 +53,15 @@ import (
 //
 // new-txn      name=<txn-name> ts=<int>[,<int>] epoch=<int> [maxts=<int>[,<int>]]
 // new-request  name=<req-name> txn=<txn-name>|none ts=<int>[,<int>] [priority] [consistency]
-//   <proto-name> [<field-name>=<field-value>...]
+//   <proto-name> [<field-name>=<field-value>...] (hint: see scanSingleRequest)
 // sequence     req=<req-name>
 // finish       req=<req-name>
 //
 // handle-write-intent-error  req=<req-name> txn=<txn-name> key=<key>
 // handle-txn-push-error      req=<req-name> txn=<txn-name> key=<key>  TODO(nvanbenschoten): implement this
 //
-// on-lock-acquired  txn=<txn-name> key=<key>
-// on-lock-updated   txn=<txn-name> key=<key> status=[committed|aborted|pending] [ts=<int>[,<int>]]
+// on-lock-acquired  req=<req-name> key=<key>
+// on-lock-updated   req=<req-name> txn=<txn-name> key=<key> status=[committed|aborted|pending] [ts=<int>[,<int>]]
 // on-txn-updated    txn=<txn-name> status=[committed|aborted|pending] [ts=<int>[,<int>]]
 //
 // on-lease-updated  leaseholder=<bool>
@@ -136,8 +136,10 @@ func TestConcurrencyManagerBasic(t *testing.T) {
 				}
 
 				ts := scanTimestamp(t, d)
-				if txn != nil && txn.ReadTimestamp != ts {
-					d.Fatalf(t, "txn read timestamp != timestamp")
+				if txn != nil {
+					txn = txn.Clone()
+					txn.ReadTimestamp = ts
+					txn.WriteTimestamp = ts
 				}
 
 				readConsistency := roachpb.CONSISTENT
@@ -149,7 +151,7 @@ func TestConcurrencyManagerBasic(t *testing.T) {
 				var reqs []roachpb.Request
 				singleReqLines := strings.Split(d.Input, "\n")
 				for _, line := range singleReqLines {
-					req := scanSingleRequest(t, d, line)
+					req := scanSingleRequest(t, d, line, c.txnsByName)
 					reqs = append(reqs, req)
 				}
 				reqUnions := make([]roachpb.RequestUnion, len(reqs))
@@ -257,18 +259,34 @@ func TestConcurrencyManagerBasic(t *testing.T) {
 				return c.waitAndCollect(t, mon)
 
 			case "on-lock-acquired":
-				var txnName string
-				d.ScanArgs(t, "txn", &txnName)
-				txn, ok := c.txnsByName[txnName]
+				var reqName string
+				d.ScanArgs(t, "req", &reqName)
+				guard, ok := c.guardsByReqName[reqName]
 				if !ok {
-					d.Fatalf(t, "unknown txn %s", txnName)
+					d.Fatalf(t, "unknown request: %s", reqName)
 				}
+				txn := guard.Req.Txn
 
 				var key string
 				d.ScanArgs(t, "key", &key)
 
+				// Confirm that the request has a corresponding write request.
+				found := false
+				for _, ru := range guard.Req.Requests {
+					req := ru.GetInner()
+					keySpan := roachpb.Span{Key: roachpb.Key(key)}
+					if roachpb.IsLocking(req) &&
+						req.Header().Span().Contains(keySpan) {
+						found = true
+						break
+					}
+				}
+				if !found {
+					d.Fatalf(t, "missing corresponding write request")
+				}
+
 				mon.runSync("acquire lock", func(ctx context.Context) {
-					log.Eventf(ctx, "%s @ %s", txnName, key)
+					log.Eventf(ctx, "txn %s @ %s", txn.ID.Short(), key)
 					span := roachpb.Span{Key: roachpb.Key(key)}
 					up := roachpb.MakeLockUpdateWithDur(txn, span, lock.Unreplicated)
 					m.OnLockAcquired(ctx, &up)
@@ -276,6 +294,13 @@ func TestConcurrencyManagerBasic(t *testing.T) {
 				return c.waitAndCollect(t, mon)
 
 			case "on-lock-updated":
+				var reqName string
+				d.ScanArgs(t, "req", &reqName)
+				guard, ok := c.guardsByReqName[reqName]
+				if !ok {
+					d.Fatalf(t, "unknown request: %s", reqName)
+				}
+
 				var txnName string
 				d.ScanArgs(t, "txn", &txnName)
 				txn, ok := c.txnsByName[txnName]
@@ -292,12 +317,27 @@ func TestConcurrencyManagerBasic(t *testing.T) {
 					ts = scanTimestamp(t, d)
 				}
 
+				// Confirm that the request has a corresponding ResolveIntent.
+				found := false
+				for _, ru := range guard.Req.Requests {
+					if riReq := ru.GetResolveIntent(); riReq != nil &&
+						riReq.IntentTxn.ID == txn.ID &&
+						riReq.Key.Equal(roachpb.Key(key)) &&
+						riReq.Status == status {
+						found = true
+						break
+					}
+				}
+				if !found {
+					d.Fatalf(t, "missing corresponding resolve intent request")
+				}
+
 				txnUpdate := txn.Clone()
 				txnUpdate.Status = status
 				txnUpdate.WriteTimestamp.Forward(ts)
 
 				mon.runSync("update lock", func(ctx context.Context) {
-					log.Eventf(ctx, "%s %s @ %s", verb, txnName, key)
+					log.Eventf(ctx, "%s txn %s @ %s", verb, txn.ID.Short(), key)
 					span := roachpb.Span{Key: roachpb.Key(key)}
 					up := roachpb.MakeLockUpdate(txnUpdate, span)
 					m.OnLockUpdated(ctx, &up)

--- a/pkg/kv/kvserver/concurrency/concurrency_manager_test.go
+++ b/pkg/kv/kvserver/concurrency/concurrency_manager_test.go
@@ -60,7 +60,7 @@ import (
 // handle-write-intent-error  req=<req-name> txn=<txn-name> key=<key>
 // handle-txn-push-error      req=<req-name> txn=<txn-name> key=<key>  TODO(nvanbenschoten): implement this
 //
-// on-lock-acquired  req=<req-name> key=<key>
+// on-lock-acquired  req=<req-name> key=<key> [seq=<seq>]
 // on-lock-updated   req=<req-name> txn=<txn-name> key=<key> status=[committed|aborted|pending] [ts=<int>[,<int>]]
 // on-txn-updated    txn=<txn-name> status=[committed|aborted|pending] [ts=<int>[,<int>]]
 //
@@ -270,13 +270,20 @@ func TestConcurrencyManagerBasic(t *testing.T) {
 				var key string
 				d.ScanArgs(t, "key", &key)
 
+				var seq int
+				if d.HasArg("seq") {
+					d.ScanArgs(t, "seq", &seq)
+				}
+				seqNum := enginepb.TxnSeq(seq)
+
 				// Confirm that the request has a corresponding write request.
 				found := false
 				for _, ru := range guard.Req.Requests {
 					req := ru.GetInner()
 					keySpan := roachpb.Span{Key: roachpb.Key(key)}
 					if roachpb.IsLocking(req) &&
-						req.Header().Span().Contains(keySpan) {
+						req.Header().Span().Contains(keySpan) &&
+						req.Header().Sequence == seqNum {
 						found = true
 						break
 					}
@@ -285,10 +292,13 @@ func TestConcurrencyManagerBasic(t *testing.T) {
 					d.Fatalf(t, "missing corresponding write request")
 				}
 
+				txnAcquire := txn.Clone()
+				txnAcquire.Sequence = seqNum
+
 				mon.runSync("acquire lock", func(ctx context.Context) {
 					log.Eventf(ctx, "txn %s @ %s", txn.ID.Short(), key)
 					span := roachpb.Span{Key: roachpb.Key(key)}
-					up := roachpb.MakeLockUpdateWithDur(txn, span, lock.Unreplicated)
+					up := roachpb.MakeLockUpdateWithDur(txnAcquire, span, lock.Unreplicated)
 					m.OnLockAcquired(ctx, &up)
 				})
 				return c.waitAndCollect(t, mon)

--- a/pkg/kv/kvserver/concurrency/datadriven_util_test.go
+++ b/pkg/kv/kvserver/concurrency/datadriven_util_test.go
@@ -57,7 +57,9 @@ func scanTimestampWithName(t *testing.T, d *datadriven.TestData, name string) hl
 	return ts
 }
 
-func scanSingleRequest(t *testing.T, d *datadriven.TestData, line string) roachpb.Request {
+func scanSingleRequest(
+	t *testing.T, d *datadriven.TestData, line string, txns map[string]*roachpb.Transaction,
+) roachpb.Request {
 	cmd, cmdArgs, err := datadriven.ParseLine(line)
 	if err != nil {
 		d.Fatalf(t, "error parsing single request: %v", err)
@@ -100,6 +102,13 @@ func scanSingleRequest(t *testing.T, d *datadriven.TestData, line string) roachp
 		r.Value.SetString(mustGetField("value"))
 		return &r
 
+	case "resolve-intent":
+		var r roachpb.ResolveIntentRequest
+		r.IntentTxn = txns[mustGetField("txn")].TxnMeta
+		r.Key = roachpb.Key(mustGetField("key"))
+		r.Status = parseTxnStatus(t, d, mustGetField("status"))
+		return &r
+
 	case "request-lease":
 		var r roachpb.RequestLeaseRequest
 		return &r
@@ -113,15 +122,31 @@ func scanSingleRequest(t *testing.T, d *datadriven.TestData, line string) roachp
 func scanTxnStatus(t *testing.T, d *datadriven.TestData) (roachpb.TransactionStatus, string) {
 	var statusStr string
 	d.ScanArgs(t, "status", &statusStr)
-	switch statusStr {
-	case "committed":
-		return roachpb.COMMITTED, "committing"
-	case "aborted":
-		return roachpb.ABORTED, "aborting"
-	case "pending":
-		return roachpb.PENDING, "increasing timestamp of"
+	status := parseTxnStatus(t, d, statusStr)
+	var verb string
+	switch status {
+	case roachpb.COMMITTED:
+		verb = "committing"
+	case roachpb.ABORTED:
+		verb = "aborting"
+	case roachpb.PENDING:
+		verb = "increasing timestamp of"
 	default:
-		d.Fatalf(t, "unknown txn statusStr: %s", statusStr)
-		return 0, ""
+		d.Fatalf(t, "unknown txn status: %s", status)
+	}
+	return status, verb
+}
+
+func parseTxnStatus(t *testing.T, d *datadriven.TestData, s string) roachpb.TransactionStatus {
+	switch s {
+	case "committed":
+		return roachpb.COMMITTED
+	case "aborted":
+		return roachpb.ABORTED
+	case "pending":
+		return roachpb.PENDING
+	default:
+		d.Fatalf(t, "unknown txn status: %s", s)
+		return 0
 	}
 }

--- a/pkg/kv/kvserver/concurrency/datadriven_util_test.go
+++ b/pkg/kv/kvserver/concurrency/datadriven_util_test.go
@@ -16,6 +16,7 @@ import (
 	"testing"
 
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/storage/enginepb"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
 	"github.com/cockroachdb/cockroach/pkg/util/uint128"
 	"github.com/cockroachdb/cockroach/pkg/util/uuid"
@@ -81,15 +82,28 @@ func scanSingleRequest(
 		}
 		return v
 	}
+	maybeGetSeq := func() enginepb.TxnSeq {
+		s, ok := fields["seq"]
+		if !ok {
+			return 0
+		}
+		n, err := strconv.ParseInt(s, 10, 64)
+		if err != nil {
+			d.Fatalf(t, "could not parse seq num: %v", err)
+		}
+		return enginepb.TxnSeq(n)
+	}
 
 	switch cmd {
 	case "get":
 		var r roachpb.GetRequest
+		r.Sequence = maybeGetSeq()
 		r.Key = roachpb.Key(mustGetField("key"))
 		return &r
 
 	case "scan":
 		var r roachpb.ScanRequest
+		r.Sequence = maybeGetSeq()
 		r.Key = roachpb.Key(mustGetField("key"))
 		if v, ok := fields["endkey"]; ok {
 			r.EndKey = roachpb.Key(v)
@@ -98,6 +112,7 @@ func scanSingleRequest(
 
 	case "put":
 		var r roachpb.PutRequest
+		r.Sequence = maybeGetSeq()
 		r.Key = roachpb.Key(mustGetField("key"))
 		r.Value.SetString(mustGetField("value"))
 		return &r

--- a/pkg/kv/kvserver/concurrency/lock_table.go
+++ b/pkg/kv/kvserver/concurrency/lock_table.go
@@ -1138,7 +1138,7 @@ func (l *lockState) acquireLock(
 		// Already held.
 		beforeTxn, beforeTs, _ := l.getLockerInfo()
 		if txn.ID != beforeTxn.ID {
-			return errors.Errorf("existing lock cannot be acquired by different transaction")
+			return errors.Errorf("caller violated contract: existing lock cannot be acquired by different transaction")
 		}
 		if l.holder.holder[durability].txn != nil && l.holder.holder[durability].txn.Epoch < txn.Epoch {
 			// Clear the sequences for the older epoch.
@@ -1168,7 +1168,7 @@ func (l *lockState) acquireLock(
 		l.holder.holder[durability].seqs = append(l.holder.holder[durability].seqs, txn.Sequence)
 		_, afterTs, _ := l.getLockerInfo()
 		if afterTs.Less(beforeTs) {
-			return errors.Errorf("caller violated contract")
+			return errors.Errorf("caller violated contract: lock timestamp regression")
 		} else if beforeTs.Less(afterTs) {
 			l.increasedLockTs(afterTs)
 		}
@@ -1266,7 +1266,7 @@ func (l *lockState) discoveredLock(
 		// the first place. Bugs here would cause infinite loops where the same
 		// lock is repeatedly re-discovered.
 		if g.readTS.Less(ts) {
-			return errors.Errorf("discovered non-conflicting lock")
+			return errors.Errorf("caller violated contract: discovered non-conflicting lock")
 		}
 
 	case spanset.SpanReadWrite:
@@ -1775,7 +1775,7 @@ func (t *lockTableImpl) AcquireLock(
 		return nil
 	}
 	if strength != lock.Exclusive {
-		return errors.Errorf("caller violated contract")
+		return errors.Errorf("caller violated contract: lock strength not Exclusive")
 	}
 	ss := spanset.SpanGlobal
 	if keys.IsLocal(key) {
@@ -1874,7 +1874,7 @@ func findAccessInSpans(
 			return sa, ss, nil
 		}
 	}
-	return 0, 0, errors.Errorf("caller violated contract")
+	return 0, 0, errors.Errorf("caller violated contract: could not find access in spans")
 }
 
 // Tries to GC locks that were previously known to have become empty.

--- a/pkg/kv/kvserver/concurrency/lock_table.go
+++ b/pkg/kv/kvserver/concurrency/lock_table.go
@@ -1140,11 +1140,11 @@ func (l *lockState) acquireLock(
 		if txn.ID != beforeTxn.ID {
 			return errors.Errorf("caller violated contract: existing lock cannot be acquired by different transaction")
 		}
+		seqs := l.holder.holder[durability].seqs
 		if l.holder.holder[durability].txn != nil && l.holder.holder[durability].txn.Epoch < txn.Epoch {
 			// Clear the sequences for the older epoch.
-			l.holder.holder[durability].seqs = l.holder.holder[durability].seqs[:0]
+			seqs = seqs[:0]
 		}
-		seqs := l.holder.holder[durability].seqs
 		if len(seqs) > 0 && seqs[len(seqs)-1] >= txn.Sequence {
 			// Idempotent lock acquisition. In this case, we simply ignore the lock
 			// acquisition as long as it corresponds to an existing sequence number.
@@ -1164,11 +1164,29 @@ func (l *lockState) acquireLock(
 			return nil
 		}
 		l.holder.holder[durability].txn = txn
-		l.holder.holder[durability].ts = ts
-		l.holder.holder[durability].seqs = append(l.holder.holder[durability].seqs, txn.Sequence)
+		// Forward the lock's timestamp instead of assigning to it blindly.
+		// While lock acquisition uses monotonically increasing timestamps
+		// from the perspective of the transaction's coordinator, this does
+		// not guarantee that a lock will never be acquired at a higher
+		// epoch and/or sequence number but with a lower timestamp when in
+		// the presence of transaction pushes. Consider the following
+		// sequence of events:
+		//
+		//  - txn A acquires lock at sequence 1, ts 10
+		//  - txn B pushes txn A to ts 20
+		//  - txn B updates lock to ts 20
+		//  - txn A's coordinator does not immediately learn of the push
+		//  - txn A re-acquires lock at sequence 2, ts 15
+		//
+		// A lock's timestamp cannot be allowed to regress, so by forwarding
+		// its timestamp during the second acquisition instead if assigning
+		// to it blindly, it remains at 20.
+		l.holder.holder[durability].ts.Forward(ts)
+		l.holder.holder[durability].seqs = append(seqs, txn.Sequence)
+
 		_, afterTs, _ := l.getLockerInfo()
 		if afterTs.Less(beforeTs) {
-			return errors.Errorf("caller violated contract: lock timestamp regression")
+			panic("lockTable bug - lock timestamp regression")
 		} else if beforeTs.Less(afterTs) {
 			l.increasedLockTs(afterTs)
 		}

--- a/pkg/kv/kvserver/concurrency/testdata/concurrency_manager/basic
+++ b/pkg/kv/kvserver/concurrency/testdata/concurrency_manager/basic
@@ -50,9 +50,9 @@ sequence req=req2
 [1] sequence req2: scanning lock table for conflicting locks
 [1] sequence req2: sequencing complete, returned guard
 
-on-lock-acquired txn=txn2 key=k
+on-lock-acquired req=req2 key=k
 ----
-[-] acquire lock: txn2 @ k
+[-] acquire lock: txn 00000002 @ k
 
 debug-lock-table
 ----
@@ -90,9 +90,20 @@ local: num=0
 # 6. Requests proceed in order
 # -------------------------------------------------------------
 
-on-lock-acquired txn=txn2 key=k
+sequence req=req2
 ----
-[-] acquire lock: txn2 @ k
+[1] sequence req2: sequencing request
+[1] sequence req2: acquiring latches
+[1] sequence req2: scanning lock table for conflicting locks
+[1] sequence req2: sequencing complete, returned guard
+
+on-lock-acquired req=req2 key=k
+----
+[-] acquire lock: txn 00000002 @ k
+
+finish req=req2
+----
+[-] finish req2: finishing request
 
 new-request name=req3 txn=txn3 ts=14,1
   get  key=k
@@ -101,20 +112,20 @@ new-request name=req3 txn=txn3 ts=14,1
 
 sequence req=req3
 ----
-[1] sequence req3: sequencing request
-[1] sequence req3: acquiring latches
-[1] sequence req3: scanning lock table for conflicting locks
-[1] sequence req3: waiting in lock wait-queues
-[1] sequence req3: pushing timestamp of txn 00000002 above 0.000000014,1
-[1] sequence req3: blocked on select in concurrency_test.(*cluster).PushTransaction
+[2] sequence req3: sequencing request
+[2] sequence req3: acquiring latches
+[2] sequence req3: scanning lock table for conflicting locks
+[2] sequence req3: waiting in lock wait-queues
+[2] sequence req3: pushing timestamp of txn 00000002 above 0.000000014,1
+[2] sequence req3: blocked on select in concurrency_test.(*cluster).PushTransaction
 
 on-txn-updated txn=txn2 status=committed
 ----
 [-] update txn: committing txn2
-[1] sequence req3: resolving intent "k" for txn 00000002 with COMMITTED status
-[1] sequence req3: acquiring latches
-[1] sequence req3: scanning lock table for conflicting locks
-[1] sequence req3: sequencing complete, returned guard
+[2] sequence req3: resolving intent "k" for txn 00000002 with COMMITTED status
+[2] sequence req3: acquiring latches
+[2] sequence req3: scanning lock table for conflicting locks
+[2] sequence req3: sequencing complete, returned guard
 
 debug-lock-table
 ----
@@ -127,9 +138,9 @@ new-request name=req4 txn=txn1 ts=10,1
 
 sequence req=req4
 ----
-[2] sequence req4: sequencing request
-[2] sequence req4: acquiring latches
-[2] sequence req4: blocked on select in spanlatch.(*Manager).waitForSignal
+[3] sequence req4: sequencing request
+[3] sequence req4: acquiring latches
+[3] sequence req4: blocked on select in spanlatch.(*Manager).waitForSignal
 
 debug-latch-manager
 ----
@@ -139,8 +150,8 @@ write count: 1
 finish req=req3
 ----
 [-] finish req3: finishing request
-[2] sequence req4: scanning lock table for conflicting locks
-[2] sequence req4: sequencing complete, returned guard
+[3] sequence req4: scanning lock table for conflicting locks
+[3] sequence req4: sequencing complete, returned guard
 
 finish req=req4
 ----
@@ -160,9 +171,20 @@ reset
 # 8. Requests proceed in order
 # -------------------------------------------------------------
 
-on-lock-acquired txn=txn2 key=k
+sequence req=req2
 ----
-[-] acquire lock: txn2 @ k
+[1] sequence req2: sequencing request
+[1] sequence req2: acquiring latches
+[1] sequence req2: scanning lock table for conflicting locks
+[1] sequence req2: sequencing complete, returned guard
+
+on-lock-acquired req=req2 key=k
+----
+[-] acquire lock: txn 00000002 @ k
+
+finish req=req2
+----
+[-] finish req2: finishing request
 
 new-request name=req5 txn=none ts=14,1
   scan key=a endkey=m
@@ -170,12 +192,12 @@ new-request name=req5 txn=none ts=14,1
 
 sequence req=req5
 ----
-[1] sequence req5: sequencing request
-[1] sequence req5: acquiring latches
-[1] sequence req5: scanning lock table for conflicting locks
-[1] sequence req5: waiting in lock wait-queues
-[1] sequence req5: pushing timestamp of txn 00000002 above 0.000000014,1
-[1] sequence req5: blocked on select in concurrency_test.(*cluster).PushTransaction
+[2] sequence req5: sequencing request
+[2] sequence req5: acquiring latches
+[2] sequence req5: scanning lock table for conflicting locks
+[2] sequence req5: waiting in lock wait-queues
+[2] sequence req5: pushing timestamp of txn 00000002 above 0.000000014,1
+[2] sequence req5: blocked on select in concurrency_test.(*cluster).PushTransaction
 
 new-request name=req6 txn=none ts=16,1
   scan key=c endkey=z
@@ -183,22 +205,22 @@ new-request name=req6 txn=none ts=16,1
 
 sequence req=req6
 ----
-[2] sequence req6: sequencing request
-[2] sequence req6: acquiring latches
-[2] sequence req6: scanning lock table for conflicting locks
-[2] sequence req6: waiting in lock wait-queues
-[2] sequence req6: blocked on select in concurrency.(*lockTableWaiterImpl).WaitOn
+[3] sequence req6: sequencing request
+[3] sequence req6: acquiring latches
+[3] sequence req6: scanning lock table for conflicting locks
+[3] sequence req6: waiting in lock wait-queues
+[3] sequence req6: blocked on select in concurrency.(*lockTableWaiterImpl).WaitOn
 
 on-txn-updated txn=txn2 status=pending ts=18,1
 ----
 [-] update txn: increasing timestamp of txn2
-[1] sequence req5: resolving intent "k" for txn 00000002 with PENDING status
-[1] sequence req5: acquiring latches
-[1] sequence req5: scanning lock table for conflicting locks
-[1] sequence req5: sequencing complete, returned guard
-[2] sequence req6: acquiring latches
-[2] sequence req6: scanning lock table for conflicting locks
-[2] sequence req6: sequencing complete, returned guard
+[2] sequence req5: resolving intent "k" for txn 00000002 with PENDING status
+[2] sequence req5: acquiring latches
+[2] sequence req5: scanning lock table for conflicting locks
+[2] sequence req5: sequencing complete, returned guard
+[3] sequence req6: acquiring latches
+[3] sequence req6: scanning lock table for conflicting locks
+[3] sequence req6: sequencing complete, returned guard
 
 new-request name=req7 txn=none ts=12,1
   put key=k value=v
@@ -206,9 +228,9 @@ new-request name=req7 txn=none ts=12,1
 
 sequence req=req7
 ----
-[3] sequence req7: sequencing request
-[3] sequence req7: acquiring latches
-[3] sequence req7: blocked on select in spanlatch.(*Manager).waitForSignal
+[4] sequence req7: sequencing request
+[4] sequence req7: acquiring latches
+[4] sequence req7: blocked on select in spanlatch.(*Manager).waitForSignal
 
 finish req=req5
 ----
@@ -217,18 +239,18 @@ finish req=req5
 finish req=req6
 ----
 [-] finish req6: finishing request
-[3] sequence req7: scanning lock table for conflicting locks
-[3] sequence req7: waiting in lock wait-queues
-[3] sequence req7: pushing txn 00000002 to abort
-[3] sequence req7: blocked on select in concurrency_test.(*cluster).PushTransaction
+[4] sequence req7: scanning lock table for conflicting locks
+[4] sequence req7: waiting in lock wait-queues
+[4] sequence req7: pushing txn 00000002 to abort
+[4] sequence req7: blocked on select in concurrency_test.(*cluster).PushTransaction
 
 on-txn-updated txn=txn2 status=committed
 ----
 [-] update txn: committing txn2
-[3] sequence req7: resolving intent "k" for txn 00000002 with COMMITTED status
-[3] sequence req7: acquiring latches
-[3] sequence req7: scanning lock table for conflicting locks
-[3] sequence req7: sequencing complete, returned guard
+[4] sequence req7: resolving intent "k" for txn 00000002 with COMMITTED status
+[4] sequence req7: acquiring latches
+[4] sequence req7: scanning lock table for conflicting locks
+[4] sequence req7: sequencing complete, returned guard
 
 finish req=req7
 ----

--- a/pkg/kv/kvserver/concurrency/testdata/concurrency_manager/deadlocks
+++ b/pkg/kv/kvserver/concurrency/testdata/concurrency_manager/deadlocks
@@ -53,17 +53,17 @@ sequence req=req3w
 [3] sequence req3w: scanning lock table for conflicting locks
 [3] sequence req3w: sequencing complete, returned guard
 
-on-lock-acquired txn=txn1 key=a
+on-lock-acquired req=req1w key=a
 ----
-[-] acquire lock: txn1 @ a
+[-] acquire lock: txn 00000001 @ a
 
-on-lock-acquired txn=txn2 key=b
+on-lock-acquired req=req2w key=b
 ----
-[-] acquire lock: txn2 @ b
+[-] acquire lock: txn 00000002 @ b
 
-on-lock-acquired txn=txn3 key=c
+on-lock-acquired req=req3w key=c
 ----
-[-] acquire lock: txn3 @ c
+[-] acquire lock: txn 00000003 @ c
 
 finish req=req1w
 ----
@@ -250,17 +250,17 @@ sequence req=req3w
 [3] sequence req3w: scanning lock table for conflicting locks
 [3] sequence req3w: sequencing complete, returned guard
 
-on-lock-acquired txn=txn1 key=a
+on-lock-acquired req=req1w key=a
 ----
-[-] acquire lock: txn1 @ a
+[-] acquire lock: txn 00000001 @ a
 
-on-lock-acquired txn=txn2 key=b
+on-lock-acquired req=req2w key=b
 ----
-[-] acquire lock: txn2 @ b
+[-] acquire lock: txn 00000002 @ b
 
-on-lock-acquired txn=txn3 key=c
+on-lock-acquired req=req3w key=c
 ----
-[-] acquire lock: txn3 @ c
+[-] acquire lock: txn 00000003 @ c
 
 finish req=req1w
 ----
@@ -474,17 +474,17 @@ sequence req=req3w
 [3] sequence req3w: scanning lock table for conflicting locks
 [3] sequence req3w: sequencing complete, returned guard
 
-on-lock-acquired txn=txn1 key=a
+on-lock-acquired req=req1w key=a
 ----
-[-] acquire lock: txn1 @ a
+[-] acquire lock: txn 00000001 @ a
 
-on-lock-acquired txn=txn2 key=b
+on-lock-acquired req=req2w key=b
 ----
-[-] acquire lock: txn2 @ b
+[-] acquire lock: txn 00000002 @ b
 
-on-lock-acquired txn=txn3 key=c
+on-lock-acquired req=req3w key=c
 ----
-[-] acquire lock: txn3 @ c
+[-] acquire lock: txn 00000003 @ c
 
 finish req=req1w
 ----
@@ -694,17 +694,17 @@ sequence req=req3w
 [3] sequence req3w: scanning lock table for conflicting locks
 [3] sequence req3w: sequencing complete, returned guard
 
-on-lock-acquired txn=txn1 key=a
+on-lock-acquired req=req1w key=a
 ----
-[-] acquire lock: txn1 @ a
+[-] acquire lock: txn 00000001 @ a
 
-on-lock-acquired txn=txn2 key=b
+on-lock-acquired req=req2w key=b
 ----
-[-] acquire lock: txn2 @ b
+[-] acquire lock: txn 00000002 @ b
 
-on-lock-acquired txn=txn3 key=c
+on-lock-acquired req=req3w key=c
 ----
-[-] acquire lock: txn3 @ c
+[-] acquire lock: txn 00000003 @ c
 
 finish req=req1w
 ----
@@ -921,17 +921,17 @@ sequence req=req3w
 [3] sequence req3w: scanning lock table for conflicting locks
 [3] sequence req3w: sequencing complete, returned guard
 
-on-lock-acquired txn=txn1 key=a
+on-lock-acquired req=req1w key=a
 ----
-[-] acquire lock: txn1 @ a
+[-] acquire lock: txn 00000001 @ a
 
-on-lock-acquired txn=txn2 key=b
+on-lock-acquired req=req2w key=b
 ----
-[-] acquire lock: txn2 @ b
+[-] acquire lock: txn 00000002 @ b
 
-on-lock-acquired txn=txn3 key=c
+on-lock-acquired req=req3w key=c
 ----
-[-] acquire lock: txn3 @ c
+[-] acquire lock: txn 00000003 @ c
 
 finish req=req1w
 ----

--- a/pkg/kv/kvserver/concurrency/testdata/concurrency_manager/range_state_listener
+++ b/pkg/kv/kvserver/concurrency/testdata/concurrency_manager/range_state_listener
@@ -61,13 +61,13 @@ sequence req=req1
 [1] sequence req1: scanning lock table for conflicting locks
 [1] sequence req1: sequencing complete, returned guard
 
-on-lock-acquired txn=txn1 key=k
+on-lock-acquired req=req1 key=k
 ----
-[-] acquire lock: txn1 @ k
+[-] acquire lock: txn 00000001 @ k
 
-on-lock-acquired txn=txn1 key=k2
+on-lock-acquired req=req1 key=k2
 ----
-[-] acquire lock: txn1 @ k2
+[-] acquire lock: txn 00000001 @ k2
 
 finish req=req1
 ----
@@ -160,9 +160,9 @@ sequence req=req2
 [6] sequence req2: scanning lock table for conflicting locks
 [6] sequence req2: sequencing complete, returned guard
 
-on-lock-acquired txn=txn2 key=k
+on-lock-acquired req=req2 key=k
 ----
-[-] acquire lock: txn2 @ k
+[-] acquire lock: txn 00000002 @ k
 
 debug-lock-table
 ----
@@ -221,10 +221,25 @@ global: num=1
    distinguished req: 3
 local: num=0
 
-on-lock-updated txn=txn2 key=k status=committed
+new-request name=reqRes2 txn=none ts=10,1
+  resolve-intent txn=txn2 key=k status=committed
 ----
-[-] update lock: committing txn2 @ k
+
+sequence req=reqRes2
+----
+[10] sequence reqRes2: sequencing request
+[10] sequence reqRes2: acquiring latches
+[10] sequence reqRes2: sequencing complete, returned guard
+
+on-lock-updated req=reqRes2 txn=txn2 key=k status=committed
+----
+[-] update lock: committing txn 00000002 @ k
 [9] sequence req3: acquiring latches
+[9] sequence req3: blocked on select in spanlatch.(*Manager).waitForSignal
+
+finish req=reqRes2
+----
+[-] finish reqRes2: finishing request
 [9] sequence req3: scanning lock table for conflicting locks
 [9] sequence req3: sequencing complete, returned guard
 
@@ -235,9 +250,9 @@ global: num=1
   res: req: 3, txn: 00000003-0000-0000-0000-000000000000, ts: 0.000000010,1, seq: 0
 local: num=0
 
-on-lock-acquired txn=txn3 key=k
+on-lock-acquired req=req3 key=k
 ----
-[-] acquire lock: txn3 @ k
+[-] acquire lock: txn 00000003 @ k
 
 debug-lock-table
 ----
@@ -293,9 +308,9 @@ sequence req=req1
 [1] sequence req1: scanning lock table for conflicting locks
 [1] sequence req1: sequencing complete, returned guard
 
-on-lock-acquired txn=txn1 key=k
+on-lock-acquired req=req1 key=k
 ----
-[-] acquire lock: txn1 @ k
+[-] acquire lock: txn 00000001 @ k
 
 finish req=req1
 ----
@@ -363,10 +378,25 @@ sequence req=req2
 [4] sequence req2: waiting in lock wait-queues
 [4] sequence req2: blocked on select in concurrency.(*lockTableWaiterImpl).WaitOn
 
-on-lock-updated txn=txn1 key=k status=committed
+new-request name=reqRes1 txn=none ts=10,1
+  resolve-intent txn=txn1 key=k status=committed
 ----
-[-] update lock: committing txn1 @ k
+
+sequence req=reqRes1
+----
+[5] sequence reqRes1: sequencing request
+[5] sequence reqRes1: acquiring latches
+[5] sequence reqRes1: sequencing complete, returned guard
+
+on-lock-updated req=reqRes1 txn=txn1 key=k status=committed
+----
+[-] update lock: committing txn 00000001 @ k
 [4] sequence req2: acquiring latches
+[4] sequence req2: blocked on select in spanlatch.(*Manager).waitForSignal
+
+finish req=reqRes1
+----
+[-] finish reqRes1: finishing request
 [4] sequence req2: scanning lock table for conflicting locks
 [4] sequence req2: sequencing complete, returned guard
 
@@ -377,9 +407,9 @@ global: num=1
   res: req: 5, txn: 00000002-0000-0000-0000-000000000000, ts: 0.000000010,1, seq: 0
 local: num=0
 
-on-lock-acquired txn=txn2 key=k
+on-lock-acquired req=req2 key=k
 ----
-[-] acquire lock: txn2 @ k
+[-] acquire lock: txn 00000002 @ k
 
 debug-lock-table
 ----
@@ -439,13 +469,13 @@ sequence req=req1
 [1] sequence req1: scanning lock table for conflicting locks
 [1] sequence req1: sequencing complete, returned guard
 
-on-lock-acquired txn=txn1 key=k
+on-lock-acquired req=req1 key=k
 ----
-[-] acquire lock: txn1 @ k
+[-] acquire lock: txn 00000001 @ k
 
-on-lock-acquired txn=txn1 key=k2
+on-lock-acquired req=req1 key=k2
 ----
-[-] acquire lock: txn1 @ k2
+[-] acquire lock: txn 00000001 @ k2
 
 finish req=req1
 ----
@@ -537,9 +567,9 @@ sequence req=req2
 [6] sequence req2: scanning lock table for conflicting locks
 [6] sequence req2: sequencing complete, returned guard
 
-on-lock-acquired txn=txn2 key=k
+on-lock-acquired req=req2 key=k
 ----
-[-] acquire lock: txn2 @ k
+[-] acquire lock: txn 00000002 @ k
 
 debug-lock-table
 ----
@@ -593,9 +623,9 @@ sequence req=req1
 [1] sequence req1: scanning lock table for conflicting locks
 [1] sequence req1: sequencing complete, returned guard
 
-on-lock-acquired txn=txn1 key=k
+on-lock-acquired req=req1 key=k
 ----
-[-] acquire lock: txn1 @ k
+[-] acquire lock: txn 00000001 @ k
 
 finish req=req1
 ----
@@ -663,10 +693,25 @@ sequence req=req2
 [4] sequence req2: waiting in lock wait-queues
 [4] sequence req2: blocked on select in concurrency.(*lockTableWaiterImpl).WaitOn
 
-on-lock-updated txn=txn1 key=k status=committed
+new-request name=reqRes1 txn=none ts=10,1
+  resolve-intent txn=txn1 key=k status=committed
 ----
-[-] update lock: committing txn1 @ k
+
+sequence req=reqRes1
+----
+[5] sequence reqRes1: sequencing request
+[5] sequence reqRes1: acquiring latches
+[5] sequence reqRes1: sequencing complete, returned guard
+
+on-lock-updated req=reqRes1 txn=txn1 key=k status=committed
+----
+[-] update lock: committing txn 00000001 @ k
 [4] sequence req2: acquiring latches
+[4] sequence req2: blocked on select in spanlatch.(*Manager).waitForSignal
+
+finish req=reqRes1
+----
+[-] finish reqRes1: finishing request
 [4] sequence req2: scanning lock table for conflicting locks
 [4] sequence req2: sequencing complete, returned guard
 
@@ -677,9 +722,9 @@ global: num=1
   res: req: 9, txn: 00000002-0000-0000-0000-000000000000, ts: 0.000000010,1, seq: 0
 local: num=0
 
-on-lock-acquired txn=txn2 key=k
+on-lock-acquired req=req2 key=k
 ----
-[-] acquire lock: txn2 @ k
+[-] acquire lock: txn 00000002 @ k
 
 debug-lock-table
 ----

--- a/pkg/kv/kvserver/concurrency/testdata/concurrency_manager/update
+++ b/pkg/kv/kvserver/concurrency/testdata/concurrency_manager/update
@@ -1,0 +1,240 @@
+# -------------------------------------------------------------
+# A transaction writes an intent. The intent is pushed to a
+# higher timestamp by a second transaction. The transaction then
+# returns to re-acquire the intent at a new sequence number but
+# still at the original timestamp. This is permitted but the
+# lock's timestamp should not regress.
+#
+# Setup: txn1 acquire lock k
+#        txn2 reads k and waits
+#        txn2 pushes txn1
+#
+# Test:  txn2 succeeds in pushing txn1's ts forward
+#        txn2 proceeds
+#        txn1 re-acquires lock k at new seq num, lower ts
+# -------------------------------------------------------------
+
+new-txn name=txn1 ts=10,1 epoch=0
+----
+
+new-txn name=txn2 ts=12,1 epoch=0
+----
+
+new-request name=req1 txn=txn1 ts=10,1
+  put key=k value=v
+----
+
+new-request name=req2 txn=txn2 ts=12,1
+  get key=k
+----
+
+sequence req=req1
+----
+[1] sequence req1: sequencing request
+[1] sequence req1: acquiring latches
+[1] sequence req1: scanning lock table for conflicting locks
+[1] sequence req1: sequencing complete, returned guard
+
+on-lock-acquired req=req1 key=k
+----
+[-] acquire lock: txn 00000001 @ k
+
+finish req=req1
+----
+[-] finish req1: finishing request
+
+sequence req=req2
+----
+[2] sequence req2: sequencing request
+[2] sequence req2: acquiring latches
+[2] sequence req2: scanning lock table for conflicting locks
+[2] sequence req2: waiting in lock wait-queues
+[2] sequence req2: pushing timestamp of txn 00000001 above 0.000000012,1
+[2] sequence req2: blocked on select in concurrency_test.(*cluster).PushTransaction
+
+debug-lock-table
+----
+global: num=1
+ lock: "k"
+  holder: txn: 00000001-0000-0000-0000-000000000000, ts: 0.000000010,1, info: unrepl epoch: 0, seqs: [0]
+   waiting readers:
+    req: 2, txn: 00000002-0000-0000-0000-000000000000
+   distinguished req: 2
+local: num=0
+
+# --------------------------------
+# Setup complete, test starts here
+# --------------------------------
+
+on-txn-updated txn=txn1 status=pending ts=12,2
+----
+[-] update txn: increasing timestamp of txn1
+[2] sequence req2: resolving intent "k" for txn 00000001 with PENDING status
+[2] sequence req2: acquiring latches
+[2] sequence req2: scanning lock table for conflicting locks
+[2] sequence req2: sequencing complete, returned guard
+
+finish req=req2
+----
+[-] finish req2: finishing request
+
+debug-lock-table
+----
+global: num=1
+ lock: "k"
+  holder: txn: 00000001-0000-0000-0000-000000000000, ts: 0.000000012,2, info: unrepl epoch: 0, seqs: [0]
+local: num=0
+
+# Issue another write to the same key for txn1 at its initial
+# timestamp. The timestamp in the lock table does not regress.
+
+new-request name=req3 txn=txn1 ts=10,1
+  put key=k value=v2 seq=1
+----
+
+sequence req=req3
+----
+[3] sequence req3: sequencing request
+[3] sequence req3: acquiring latches
+[3] sequence req3: scanning lock table for conflicting locks
+[3] sequence req3: sequencing complete, returned guard
+
+on-lock-acquired req=req3 key=k seq=1
+----
+[-] acquire lock: txn 00000001 @ k
+
+finish req=req3
+----
+[-] finish req3: finishing request
+
+debug-lock-table
+----
+global: num=1
+ lock: "k"
+  holder: txn: 00000001-0000-0000-0000-000000000000, ts: 0.000000012,2, info: unrepl epoch: 0, seqs: [0, 1]
+local: num=0
+
+reset namespace
+----
+
+# -------------------------------------------------------------
+# A transaction writes an intent. The intent is pushed to a
+# higher timestamp by a second transaction. The transaction then
+# returns to re-acquire the intent at a new epoch but still at
+# the original timestamp. This is permitted but the lock's
+# timestamp should not regress.
+#
+# Setup: txn1 acquire lock k
+#        txn2 reads k and waits
+#
+# Test:  txn2 pushes txn1's timestamp forward
+#        txn2 proceeds
+#        txn1 re-acquires lock k at new epoch, lower ts
+# -------------------------------------------------------------
+
+new-txn name=txn1 ts=10,1 epoch=0
+----
+
+new-txn name=txn2 ts=12,1 epoch=0
+----
+
+new-request name=req1 txn=txn1 ts=10,1
+  put key=k value=v
+----
+
+new-request name=req2 txn=txn2 ts=12,1
+  get key=k
+----
+
+sequence req=req1
+----
+[1] sequence req1: sequencing request
+[1] sequence req1: acquiring latches
+[1] sequence req1: scanning lock table for conflicting locks
+[1] sequence req1: sequencing complete, returned guard
+
+on-lock-acquired req=req1 key=k
+----
+[-] acquire lock: txn 00000001 @ k
+
+finish req=req1
+----
+[-] finish req1: finishing request
+
+sequence req=req2
+----
+[2] sequence req2: sequencing request
+[2] sequence req2: acquiring latches
+[2] sequence req2: scanning lock table for conflicting locks
+[2] sequence req2: waiting in lock wait-queues
+[2] sequence req2: pushing timestamp of txn 00000001 above 0.000000012,1
+[2] sequence req2: blocked on select in concurrency_test.(*cluster).PushTransaction
+
+debug-lock-table
+----
+global: num=1
+ lock: "k"
+  holder: txn: 00000001-0000-0000-0000-000000000000, ts: 0.000000010,1, info: unrepl epoch: 0, seqs: [0]
+   waiting readers:
+    req: 5, txn: 00000002-0000-0000-0000-000000000000
+   distinguished req: 5
+local: num=0
+
+# --------------------------------
+# Setup complete, test starts here
+# --------------------------------
+
+on-txn-updated txn=txn1 status=pending ts=12,2
+----
+[-] update txn: increasing timestamp of txn1
+[2] sequence req2: resolving intent "k" for txn 00000001 with PENDING status
+[2] sequence req2: acquiring latches
+[2] sequence req2: scanning lock table for conflicting locks
+[2] sequence req2: sequencing complete, returned guard
+
+finish req=req2
+----
+[-] finish req2: finishing request
+
+debug-lock-table
+----
+global: num=1
+ lock: "k"
+  holder: txn: 00000001-0000-0000-0000-000000000000, ts: 0.000000012,2, info: unrepl epoch: 0, seqs: [0]
+local: num=0
+
+# The txn restarts at a new timestamp, but below the pushed
+# timestamp. It re-issues the same write at the new epoch. The
+# timestamp in the lock table does not regress.
+
+new-txn name=txn1 ts=11,1 epoch=1
+----
+
+new-request name=req3 txn=txn1 ts=11,1
+  put key=k value=v2
+----
+
+sequence req=req3
+----
+[3] sequence req3: sequencing request
+[3] sequence req3: acquiring latches
+[3] sequence req3: scanning lock table for conflicting locks
+[3] sequence req3: sequencing complete, returned guard
+
+on-lock-acquired req=req3 key=k
+----
+[-] acquire lock: txn 00000001 @ k
+
+finish req=req3
+----
+[-] finish req3: finishing request
+
+debug-lock-table
+----
+global: num=1
+ lock: "k"
+  holder: txn: 00000001-0000-0000-0000-000000000000, ts: 0.000000012,2, info: unrepl epoch: 1, seqs: [0]
+local: num=0
+
+reset namespace
+----

--- a/pkg/kv/kvserver/concurrency/testdata/lock_table/lock_changes
+++ b/pkg/kv/kvserver/concurrency/testdata/lock_table/lock_changes
@@ -31,16 +31,35 @@ global: num=1
 local: num=0
 
 # ---------------------------------------------------------------------------------
+# Lock is reacquired at same epoch with lower timestamp. This is allowed,
+# see TestMVCCHistories/put_out_of_order. The new sequence number is added
+# but the timestamp is not regressed.
+# ---------------------------------------------------------------------------------
+
+new-txn txn=txn1 ts=8 epoch=0 seq=3
+----
+
+new-request r=req2 txn=txn1 ts=8 spans=w@a
+----
+
+acquire r=req2 k=a durability=u
+----
+global: num=1
+ lock: "a"
+  holder: txn: 00000000-0000-0000-0000-000000000001, ts: 0.000000010,0, info: unrepl epoch: 0, seqs: [2, 3]
+local: num=0
+
+# ---------------------------------------------------------------------------------
 # Lock is reacquired at a different epoch. The old sequence numbers are discarded.
 # ---------------------------------------------------------------------------------
 
 new-txn txn=txn1 ts=10 epoch=1 seq=0
 ----
 
-new-request r=req2 txn=txn1 ts=10 spans=w@a
+new-request r=req3 txn=txn1 ts=10 spans=w@a
 ----
 
-acquire r=req2 k=a durability=u
+acquire r=req3 k=a durability=u
 ----
 global: num=1
  lock: "a"
@@ -48,17 +67,36 @@ global: num=1
 local: num=0
 
 # ---------------------------------------------------------------------------------
+# Lock is reacquired at a different epoch with lower timestamp. This is allowed,
+# see TestMVCCHistories/put_new_epoch_lower_timestamp. The old sequence numbers are
+# discarded but the timestamp is not regressed.
+# ---------------------------------------------------------------------------------
+
+new-txn txn=txn1 ts=8 epoch=2 seq=0
+----
+
+new-request r=req4 txn=txn1 ts=8 spans=w@a
+----
+
+acquire r=req4 k=a durability=u
+----
+global: num=1
+ lock: "a"
+  holder: txn: 00000000-0000-0000-0000-000000000001, ts: 0.000000010,0, info: unrepl epoch: 2, seqs: [0]
+local: num=0
+
+# ---------------------------------------------------------------------------------
 # Reader waits until the timestamp of the lock is updated.
 # ---------------------------------------------------------------------------------
 
-new-request r=req3 txn=txn2 ts=12 spans=r@a
+new-request r=req5 txn=txn2 ts=12 spans=r@a
 ----
 
-scan r=req3
+scan r=req5
 ----
 start-waiting: true
 
-guard-state r=req3
+guard-state r=req5
 ----
 new: state=waitForDistinguished txn=txn1 ts=10 key="a" held=true guard-access=read
 
@@ -66,7 +104,7 @@ print
 ----
 global: num=1
  lock: "a"
-  holder: txn: 00000000-0000-0000-0000-000000000001, ts: 0.000000010,0, info: unrepl epoch: 1, seqs: [0]
+  holder: txn: 00000000-0000-0000-0000-000000000001, ts: 0.000000010,0, info: unrepl epoch: 2, seqs: [0]
    waiting readers:
     req: 1, txn: 00000000-0000-0000-0000-000000000002
    distinguished req: 1
@@ -75,17 +113,17 @@ local: num=0
 new-txn txn=txn1 ts=14 epoch=1 seq=1
 ----
 
-new-request r=req4 txn=txn1 ts=14 spans=w@a
+new-request r=req6 txn=txn1 ts=14 spans=w@a
 ----
 
-acquire r=req4 k=a durability=u
+acquire r=req6 k=a durability=u
 ----
 global: num=1
  lock: "a"
   holder: txn: 00000000-0000-0000-0000-000000000001, ts: 0.000000014,0, info: unrepl epoch: 1, seqs: [0, 1]
 local: num=0
 
-guard-state r=req3
+guard-state r=req5
 ----
 new: state=doneWaiting
 
@@ -96,18 +134,18 @@ new: state=doneWaiting
 # that the lock table is used.
 # ---------------------------------------------------------------------------------
 
-new-request r=req5 txn=txn2 ts=17 spans=r@a
+new-request r=req7 txn=txn2 ts=17 spans=r@a
 ----
 
-scan r=req5
+scan r=req7
 ----
 start-waiting: true
 
-guard-state r=req5
+guard-state r=req7
 ----
 new: state=waitForDistinguished txn=txn1 ts=14 key="a" held=true guard-access=read
 
-add-discovered r=req5 k=a txn=txn1
+add-discovered r=req7 k=a txn=txn1
 ----
 global: num=1
  lock: "a"
@@ -117,6 +155,6 @@ global: num=1
    distinguished req: 2
 local: num=0
 
-guard-state r=req5
+guard-state r=req7
 ----
 new: state=waitForDistinguished txn=txn1 ts=14 key="a" held=true guard-access=read

--- a/pkg/storage/testdata/mvcc_histories/put_new_epoch_lower_sequence
+++ b/pkg/storage/testdata/mvcc_histories/put_new_epoch_lower_sequence
@@ -4,10 +4,6 @@
 # from the old epoch.
 
 # Additionally the intent history is blown away when a transaction restarts.
-#
-# TODO(knz): This last sentence is dubious. The test, as-is, does not
-# create an intent history because there is just 1 put.
-# See: https://github.com/cockroachdb/cockroach/issues/42310
 
 # Populate a txn and make the intent have an intent history by using
 # two puts.
@@ -15,15 +11,17 @@
 run ok
 with t=A
   txn_begin ts=1
-  txn_step  n=5
+  txn_step  n=4
   put  k=k v=v
+  txn_step
+  put  k=k v=v2
   get  k=k ts=3
 ----
-get: "k" -> /BYTES/v @0.000000001,0
+get: "k" -> /BYTES/v2 @0.000000001,0
 >> at end:
 txn: "A" meta={id=00000000 key=/Min pri=0.00000000 epo=0 ts=0.000000001,0 min=0,0 seq=5} lock=true stat=PENDING rts=0.000000001,0 wto=false max=0,0
-meta: "k"/0,0 -> txn={id=00000000 key=/Min pri=0.00000000 epo=0 ts=0.000000001,0 min=0,0 seq=5} ts=0.000000001,0 del=false klen=12 vlen=6
-data: "k"/0.000000001,0 -> /BYTES/v
+meta: "k"/0,0 -> txn={id=00000000 key=/Min pri=0.00000000 epo=0 ts=0.000000001,0 min=0,0 seq=5} ts=0.000000001,0 del=false klen=12 vlen=7 ih={{4 /BYTES/v}}
+data: "k"/0.000000001,0 -> /BYTES/v2
 
 run ok
 with t=A
@@ -38,25 +36,21 @@ txn: "A" meta={id=00000000 key=/Min pri=0.00000000 epo=1 ts=0.000000001,0 min=0,
 
 # The following will blow the intent history because the epoch is now
 # higher. However the intent is preserved.
-#
-# TODO(knz): This last sentence is dubious. The test, as-is, does not
-# create an intent history because there is just 1 put.
-# See: https://github.com/cockroachdb/cockroach/issues/42310
 
 run ok
 with t=A k=k
-  put v=v2
+  put v=v3
   check_intent exists
 ----
 meta: "k" -> txn={id=00000000 key=/Min pri=0.00000000 epo=1 ts=0.000000001,0 min=0,0 seq=4} ts=0.000000001,0 del=false klen=12 vlen=7
 >> at end:
 meta: "k"/0,0 -> txn={id=00000000 key=/Min pri=0.00000000 epo=1 ts=0.000000001,0 min=0,0 seq=4} ts=0.000000001,0 del=false klen=12 vlen=7
-data: "k"/0.000000001,0 -> /BYTES/v2
+data: "k"/0.000000001,0 -> /BYTES/v3
 
-# We're expecting v2 here.
+# We're expecting v3 here.
 
 run ok
 with t=A
   get k=k
 ----
-get: "k" -> /BYTES/v2 @0.000000001,0
+get: "k" -> /BYTES/v3 @0.000000001,0

--- a/pkg/storage/testdata/mvcc_histories/put_new_epoch_lower_timestamp
+++ b/pkg/storage/testdata/mvcc_histories/put_new_epoch_lower_timestamp
@@ -1,0 +1,46 @@
+# Tests a scenario where a put operation of an older timestamp but a
+# higher epoch comes after a put operation of a newer timestamp but an
+# older epoch. The timestamp of the resulting intent remains equal to
+# the higher timestamp - it does not regress.
+#
+# Transaction coordinators shouldn't issue this series of operations
+# directly, but it is possible to create such a situation if the
+# transaction is pushed.
+
+# Populate a txn and an intent.
+
+run ok
+with t=A
+  txn_begin   ts=1
+  txn_advance ts=5
+  txn_step    n=4
+  put  k=k v=v
+  get  k=k ts=3
+----
+get: "k" -> /BYTES/v @0.000000005,0
+>> at end:
+txn: "A" meta={id=00000000 key=/Min pri=0.00000000 epo=0 ts=0.000000005,0 min=0,0 seq=4} lock=true stat=PENDING rts=0.000000001,0 wto=false max=0,0
+meta: "k"/0,0 -> txn={id=00000000 key=/Min pri=0.00000000 epo=0 ts=0.000000005,0 min=0,0 seq=4} ts=0.000000005,0 del=false klen=12 vlen=6
+data: "k"/0.000000005,0 -> /BYTES/v
+
+run ok
+with t=A
+  txn_advance ts=3
+  txn_restart
+----
+>> at end:
+txn: "A" meta={id=00000000 key=/Min pri=0.00000000 epo=1 ts=0.000000003,0 min=0,0 seq=0} lock=true stat=PENDING rts=0.000000003,0 wto=false max=0,0
+
+
+# We're operating at a higher epoch but a lower timestamp.
+# We're expecting v2 here, but still at timestamp 5.
+
+run ok
+with t=A
+  put k=k v=v2
+  get k=k
+----
+get: "k" -> /BYTES/v2 @0.000000005,0
+>> at end:
+meta: "k"/0,0 -> txn={id=00000000 key=/Min pri=0.00000000 epo=1 ts=0.000000003,0 min=0,0 seq=0} ts=0.000000005,0 del=false klen=12 vlen=7
+data: "k"/0.000000005,0 -> /BYTES/v2

--- a/pkg/storage/testdata/mvcc_histories/put_out_of_order
+++ b/pkg/storage/testdata/mvcc_histories/put_out_of_order
@@ -1,5 +1,7 @@
-# tests a scenario where a put operation of an
-# older timestamp comes after a put operation of a newer timestamp.
+# Tests a scenario where a put operation of an older timestamp comes
+# after a put operation of a newer timestamp. The timestamp of the
+# resulting intent remains equal to the higher timestamp - it does not
+# regress.
 
 run ok
 with t=A

